--- libfoo.pl_orig	2024-12-08 19:58:09.314195900 +0100
+++ libfoo.pl	2024-12-15 20:17:44.702538879 +0100
@@ -17,7 +17,7 @@
 my $router = $ENV{"SRCBASE"} . "/router";
 my $openssldir = $ENV{"OPENSSLDIR"};
 my $is_arm = $ENV{"CONFIG_BCMWL6A"};
-my $uclibc_lib = $ENV{"TOOLCHAIN"} . "/lib";
+my $uclibc_lib = $ENV{"TOOLCHAIN"} . "/mipsel-brcm-linux-uclibc/sysroot/usr/lib";
 my $uclibc_bin = $ENV{"TOOLCHAIN"} . "/bin";
 my $stripshared;
 
@@ -566,7 +566,7 @@
 		$cmd = $uclibc_bin . "/arm-brcm-linux-uclibcgnueabi-gcc -shared -nostdlib -Wl,-s,-z,combreloc -Wl,--warn-common -Wl,--gc-sections ${opt} -Wl,-soname=${name} -o ${so}";
 	}
 	else {
-		$cmd = $uclibc_bin . "/mipsel-uclibc-gcc -shared -nostdlib -Wl,-s,-z,combreloc -Wl,--warn-common -Wl,--fatal-warnings -Wl,--gc-sections ${opt} -Wl,-soname=${name} -o ${so}";
+		$cmd = $uclibc_bin . "/mipsel-linux-gcc -shared -nostdlib -Wl,-s,-z,combreloc -Wl,--warn-common -Wl,--gc-sections ${opt} -Wl,-soname=${name} -o ${so}";
 	}
 
 	foreach (@{$elf_lib{$name}}) {
@@ -638,7 +638,7 @@
 }
 
 if ($is_arm ne "y") { # MIPS only
-	genSO("${root}/lib/libc.so.0", "${uclibc_lib}/libc.a", "", "-Wl,-init=__uClibc_init ${uclibc_lib}/optinfo/interp.os");
+	genSO("${root}/lib/libc.so.0", "${uclibc_lib}/libc.so", "", "-Wl,-init=__uClibc_init ${uclibc_lib}/interp.os");
 	genSO("${root}/lib/libresolv.so.0", "${uclibc_lib}/libresolv.a", "${stripshared}");
 	genSO("${root}/lib/libcrypt.so.0", "${uclibc_lib}/libcrypt.a", "${stripshared}");
 	genSO("${root}/lib/libm.so.0", "${uclibc_lib}/libm.a");
