--- configure.ac	2020-01-25 18:49:53.560342000 +0100
+++ configure.ac_mod	2020-01-25 19:48:13.903733587 +0100
@@ -160,10 +160,10 @@
         have_systemd=no)
     if test "x$have_systemd" = "xno"; then
         AC_MSG_NOTICE([Okay, checking for systemd a different way...])
-        PKG_CHECK_MODULES(SYSTEMD,
-            [libsystemd],
-            have_systemd=yes,
-            have_systemd=no)
+#        PKG_CHECK_MODULES(SYSTEMD,
+#            [libsystemd],
+#            have_systemd=yes,
+#            have_systemd=no)
     fi
 fi
 

