--- Makefile.orig	2021-11-05 19:05:24.778702200 +0100
+++ Makefile.mod	2021-11-05 19:22:07.768702200 +0100
@@ -1140,7 +1140,7 @@
 libnfnetlink: libnfnetlink/stamp-h1
 	@$(SEP)
 	$(MAKE) -C libnfnetlink $(PARALLEL_BUILD)
-	@rm -f libnfnetlink/src/libnfnetlink.la
+	sed "s|'/usr/lib|'$(TOP)/libnfnetlink/src|g" -i $(TOP)/libnfnetlink/src/libnfnetlink.la
 
 libnfnetlink-install:
 	install -D libnfnetlink/src/.libs/libnfnetlink.so.0.2.0 $(INSTALLDIR)/libnfnetlink/usr/lib/libnfnetlink.so.0.2.0
@@ -1992,8 +1992,9 @@
 	@$(SEP)
 	@$(MAKE) -C libpng all $(PARALLEL_BUILD)
 	@$(MAKE) -C libpng DESTDIR=$(TOP)/libpng/staged install
-	@rm -f $(TOP)/libpng/staged/usr/lib/libpng.la
-	@rm -f $(TOP)/libpng/staged/usr/lib/libpng12.la
+	sed "s|'/usr/lib|'$(TOP)/libpng/staged/usr/lib|g" -i $(TOP)/libpng/staged/usr/lib/libpng.la
+	sed "s|'/usr/lib|'$(TOP)/libpng/staged/usr/lib|g" -i $(TOP)/libpng/staged/usr/lib/libpng12.la
+
 
 libpng-install:
 	install -D libpng/.libs/libpng16.so.16.37.0 $(INSTALLDIR)/libpng/usr/lib/libpng16.so.16.37.0
@@ -2943,7 +2944,7 @@
 	@$(SEP)
 	$(MAKE) -C libmnl
 	$(MAKE) -C libmnl DESTDIR=$(TOP)/libmnl/staged install
-	rm -f $(TOP)/libmnl/staged/usr/lib/libmnl.la
+	sed "s|'/usr/lib|'$(TOP)/libmnl/staged/usr/lib|g" -i $(TOP)/libmnl/staged/usr/lib/libmnl.la
 
 libmnl-install:
 	install -d $(INSTALLDIR)/libmnl/usr/lib/
@@ -2970,7 +2971,7 @@
 	@$(SEP)
 	$(MAKE) -C libnetfilter_conntrack
 	$(MAKE) -C libnetfilter_conntrack DESTDIR=$(TOP)/libnetfilter_conntrack/staged install
-	@rm -f libnetfilter_conntrack/staged/usr/lib/libnetfilter_conntrack.la
+	sed "s|'/usr/lib|'$(TOP)/libnetfilter_conntrack/staged/usr/lib|g" -i $(TOP)/libnetfilter_conntrack/staged/usr/lib/libnetfilter_conntrack.la
 
 libnetfilter_conntrack-install:
 	install -d $(INSTALLDIR)/libnetfilter_conntrack/usr/lib/
