--- Makefile	2024-09-13 13:21:53.957772700 +0200
+++ Makefile_mod	2024-09-13 12:20:43.967896541 +0200
@@ -1855,6 +1854,7 @@
 	@$(SEP)
 	@$(MAKE) -C $@ all $(PARALLEL_BUILD)
 	@$(MAKE) -C $@ DESTDIR=$(TOP)/sqlite/staged install
+	sed "s|'/usr/lib|'$(TOP)/sqlite/staged/usr/lib|g" -i $(TOP)/sqlite/staged/usr/lib/libsqlite3.la
 
 sqlite-install:
 ifeq ($(or $(TCONFIG_NGINX),$(TCONFIG_MEDIA_SERVER)),y)
@@ -3788,6 +3788,7 @@
 	@$(SEP)
 	@$(MAKE) -C $@ $(PARALLEL_BUILD)
 	@$(MAKE) -C $@ DESTDIR=$(TOP)/expat/staged install
+	sed "s|'/usr/lib|'$(TOP)/expat/staged/usr/lib|g" -i $(TOP)/expat/staged/usr/lib/libexpat.la
 
 expat-install:
 	install -D expat/staged/usr/lib/libexpat.so.1.9.2 $(INSTALLDIR)/expat/usr/lib/libexpat.so.1.9.2
