diff -ruN a/cmd/ztest/ztest.c b/cmd/ztest/ztest.c
--- a/cmd/ztest/ztest.c	2022-08-06 11:17:03.027079090 +0200
+++ b/cmd/ztest/ztest.c	2022-08-05 18:33:02.351781000 +0200
@@ -134,7 +134,7 @@
 #include <libzutil.h>
 #include <sys/crypto/icp.h>
 #ifdef __GLIBC__
-#include <execinfo.h> /* for backtrace() */
+/*#include <execinfo.h> /* for backtrace() */
 #endif
 
 static int ztest_fd_data = -1;
@@ -560,11 +560,11 @@
 {
 	struct sigaction action;
 #ifdef __GLIBC__ /* backtrace() is a GNU extension */
-	int nptrs;
+/*	int nptrs;
 	void *buffer[BACKTRACE_SZ];
 
 	nptrs = backtrace(buffer, BACKTRACE_SZ);
-	backtrace_symbols_fd(buffer, nptrs, STDERR_FILENO);
+	backtrace_symbols_fd(buffer, nptrs, STDERR_FILENO); */
 #endif
 	dump_debug_buffer();
 
diff -ruN a/tests/zfs-tests/cmd/xattrtest/xattrtest.c b/tests/zfs-tests/cmd/xattrtest/xattrtest.c
--- a/tests/zfs-tests/cmd/xattrtest/xattrtest.c	2022-08-06 11:17:03.427084384 +0200
+++ b/tests/zfs-tests/cmd/xattrtest/xattrtest.c	2022-08-05 18:24:28.867941000 +0200
@@ -44,7 +44,7 @@
 #include <sys/time.h>
 #include <linux/limits.h>
 
-extern char *program_invocation_short_name;
+extern const char *program_invocation_short_name;
 
 #define	ERROR(fmt, ...)                                                 \
 	fprintf(stderr, "%s: %s:%d: %s: " fmt "\n",                     \
