--- common.mak.orig	2022-02-11 16:40:59.128827400 +0100
+++ common.mak	2022-02-11 17:11:50.768827400 +0100
@@ -97,8 +97,8 @@
 endif
 endif # TCONFIG_BCMARM
 
-export LIBDIR := $(TOOLCHAIN)/lib
-export USRLIBDIR := $(TOOLCHAIN)/usr/lib
+export LIBDIR := $(shell cd $(dir $(shell which ld-uClibc-0.9.32.1.so)) && pwd -P)
+export USRLIBDIR := $(shell cd $(dir $(shell which ld-uClibc-0.9.32.1.so))/../usr/lib && pwd -P)
 
 export PLATFORMDIR := $(TOP)/$(PLATFORM)
 export INSTALLDIR := $(PLATFORMDIR)/install
