--- a/release/src-rt-6.x/shared/bzip2_inflate.c	2022-01-24 14:05:37.466249900 +0100
+++ b/gcc-5.3-toolchain_mips/bzip2_inflate.c	2022-01-03 20:14:54.000000000 +0100
@@ -95,7 +95,7 @@
 #endif
 
 extern void bz_internal_error ( int errcode );
-__inline__ Int32 BZ2_indexIntoF ( Int32 indx, Int32 *cftab );
+static inline Int32 BZ2_indexIntoF ( Int32 indx, Int32 *cftab );
 void BZ2_hbCreateDecodeTables ( Int32 *limit,
                                 Int32 *base,
                                 Int32 *perm,
@@ -845,7 +848,7 @@
 }
 
 /*---------------------------------------------------*/
-__inline__ Int32 BZ2_indexIntoF ( Int32 indx, Int32 *cftab )
+static inline Int32 BZ2_indexIntoF ( Int32 indx, Int32 *cftab )
 {
    Int32 nb, na, mid;
    nb = 0;
