--- a/release/src/router/common.mak	2022-01-24 14:05:37.476249900 +0100
+++ b/gcc-5.3-toolchain_mips/common.mak	2022-01-01 12:30:45.000000000 +0100
@@ -32,31 +21,14 @@
 export BUILD := $(shell (gcc -dumpmachine))
 export HOSTCC := gcc
 
-ifeq ($(TCONFIG_BCMARM),y)
-export PLATFORM := arm-uclibc
-export CROSS_COMPILE := arm-brcm-linux-uclibcgnueabi-
-export CROSS_COMPILER := $(CROSS_COMPILE)
-export CONFIGURE := ./configure --host=arm-linux --build=$(BUILD)
-export HOSTCONFIG := linux-armv4
-ifeq ($(CONFIG_BCM7),y)
-export BCMEX := _arm_7
-else
-export BCMEX := _arm
-endif
-export EXTRA_FLAG := -lgcc_s
-export ARCH := arm
-export HOST := arm-linux
-else
 export PLATFORM := mipsel-uclibc
-export CROSS_COMPILE := mipsel-uclibc-
+export CROSS_COMPILE := mipsel-linux-
 export CROSS_COMPILER := $(CROSS_COMPILE)
 export CONFIGURE := ./configure --host=mipsel-linux --build=$(BUILD)
 export HOSTCONFIG := linux-mipsel
 export ARCH := mips
 export HOST := mipsel-linux
-endif
 
-export PLT := $(ARCH)
 export TOOLCHAIN := $(shell cd $(dir $(shell which $(CROSS_COMPILE)strip))/.. && pwd -P)
 
 export CC := $(CROSS_COMPILE)gcc
