--- a/release/src/router/iptables/ip6tables.c	2022-01-24 14:05:37.476249900 +0100
+++ b/gcc-5.3-toolchain_mips/ip6tables.c	2022-01-04 11:36:18.000000000 +0100
@@ -730,7 +730,7 @@
 	for (i = 0, j = 0; i < n; i++) {
 		int k;
 		for (k = 0; k < 4; k++)
-			addrp[j].in6_u.u6_addr32[k] &= maskp->in6_u.u6_addr32[k];
+			addrp[j].s6_addr32[k] &= maskp->s6_addr32[k];
 		j++;
 		for (k = 0; k < j - 1; k++) {
 			if (IN6_ARE_ADDR_EQUAL(&addrp[k], &addrp[j - 1])) {
