--- ip6tables-standalone.c_orig	2022-01-27 15:12:09.000000000 +0000
+++ ip6tables-standalone.c	2022-01-04 12:22:19.000000000 +0000
@@ -34,10 +34,12 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <errno.h>
-#include <ip6tables.h>
+#include "libiptc/libip6tc.h"
+#include "ip6tables.h"
+
 
 int
-main(int argc, char *argv[])
+ip6tables_main(int argc, char *argv[])
 {
 	int ret;
 	char *table = "filter";
