--- a/release/src/router/iptables/ip6tables-standalone.c	2022-01-24 14:05:37.476249900 +0100
+++ b/gcc-5.3-toolchain_mips/ip6tables-standalone.c	2022-01-04 13:22:19.000000000 +0100
@@ -34,10 +34,12 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <errno.h>
-#include <ip6tables.h>
+#include "libiptc/libip6tc.h"
+#include "ip6tables.h"
+
 
 int
-main(int argc, char *argv[])
+ip6tables_main(int argc, char *argv[])
 {
 	int ret;
 	char *table = "filter";
