--- a/release/src/btools/libfoo.pl	2022-01-24 14:05:37.466249900 +0100
+++ b/gcc-5.3-toolchain_mips/libfoo_mips2.pl	2022-01-08 16:48:12.906611200 +0100
@@ -140,9 +140,15 @@
 	foreach (@elfs) {
 		if (/^libipt_.+\.so$/) {
 			fixDynDep("iptables", $_);
+			fixDynDep($_, "libxtables.so");
 		}
 		elsif (/^libip6t_.+\.so$/) {
 			fixDynDep("ip6tables", $_);
+			fixDynDep($_, "libxtables.so");
+		}
+		elsif (/^libxt_.+\.so$/) {
+			fixDynDep("ip6tables", $_);
+			fixDynDep($_, "libxtables.so");
 		}
 		elsif (/^CP\d+\.so$/) {
 			fixDynDep("smbd", $_);
@@ -374,7 +380,7 @@
 	my $t;
 	my $found;
 
-#	print "Resolving implicit links...\n";
+	print LOG "Resolving implicit links...\n";
 	
 	foreach $name (@elfs) {
 		foreach $sym (keys %{$elf_ext{$name}}) {
@@ -439,7 +445,7 @@
 	my $s;
 	my @u;
 	
-#	print "Generating Xref Report...\n";
+	print "Generating Xref Report...\n";
 	
 	open($f, ">libfoo_xref.txt");
 	foreach $fname (sort keys %elf_type) {
@@ -514,8 +520,9 @@
 
 	print LOG "\n\n${base}\n";
 	
-#	$cmd = "mipsel-uclibc-ld -shared -s -z combreloc --warn-common --fatal-warnings ${opt} -soname ${name} -o ${so}";
-	$cmd = "mipsel-uclibc-gcc -shared -nostdlib -Wl,-s,-z,combreloc -Wl,--warn-common -Wl,--fatal-warnings -Wl,--gc-sections ${opt} -Wl,-soname=${name} -o ${so}";
+#	$cmd = "mipsel-brcm-linux-uclibc-ld -shared -s -z combreloc --warn-common --fatal-warnings ${opt} -soname ${name} -o ${so}";
+#	$cmd = "mipsel-brcm-linux-uclibc-gcc -shared -nostdlib -Wl,-s,-z,combreloc -Wl,--warn-common -Wl,--fatal-warnings -Wl,--gc-sections ${opt} -Wl,-soname=${name} -o ${so}";
+	$cmd = "mipsel-brcm-linux-uclibc-gcc -shared -nostdlib -Wl,-s,-z,combreloc -Wl,--warn-common -Wl,--gc-sections ${opt} -Wl,-soname=${name} -o ${so}";
 	foreach (@{$elf_lib{$name}}) {
 		if ((!$elf_dyn{$name}{$_}) && (/^lib(.+)\.so/)) {
 			$cmd .= " -l$1";
@@ -526,7 +533,7 @@
 	}
 #	print "$cmd -u... ${arc}\n";	
 	if (scalar(@used) == 0) {
-		print "$name: WARNING: Library is not used by anything, deleting...\n";
+		print "$name: WARNING: Library is not used by anything, will be deleted...\n";
 		unlink $so;
 #		<>;
 		return 0;
@@ -551,7 +558,6 @@
 	
 #	print "\n$name: Attempting to link ", scalar(@used), " and remove ", scalar(@unused), " objects...\n";
 #	printf "Before: %.2fK / After: %.2fK / Removed: %.2fK\n\n", $before / 1024, $after / 1024, ($before - $after) / 1024;
-
 	return ($before > $after)
 }
 
@@ -568,12 +574,12 @@
 	exit(1);
 }
 
-#open(LOG, ">libfoo.debug");
-open(LOG, ">/dev/null");
+open(LOG, ">libfoo.debug");
+#open(LOG, ">/dev/null");
 
-print "\r--- Loading...\r\r";
+print "Loading...\r";
 load($root);
-print "\r--- Finished loading files. ---\r\r";
+print "Finished loading files.", " " x 30, "\r";
 
 fixDyn();
 fillGaps();
@@ -584,29 +590,28 @@
 if ($ARGV[0] eq "--noopt") {
 	$stripshared = "no";
 }
-
-genSO("${root}/lib/libc.so.0", "${uclibc}/lib/libc.a", "", "-Wl,-init=__uClibc_init ${uclibc}/lib/optinfo/interp.os");
-genSO("${root}/lib/libresolv.so.0", "${uclibc}/lib/libresolv.a", "${stripshared}");
-genSO("${root}/lib/libcrypt.so.0", "${uclibc}/lib/libcrypt.a", "${stripshared}");
-genSO("${root}/lib/libm.so.0", "${uclibc}/lib/libm.a");
-genSO("${root}/lib/libpthread.so.0", "${uclibc}/lib/libpthread.a", "${stripshared}", "-u pthread_mutexattr_init -Wl,-init=__pthread_initialize_minimal_internal");
-genSO("${root}/lib/libutil.so.0", "${uclibc}/lib/libutil.a", "${stripshared}");
-#genSO("${root}/lib/libdl.so.0", "${uclibc}/lib/libdl.a", "${stripshared}");
-#genSO("${root}/lib/libnsl.so.0", "${uclibc}/lib/libnsl.a", "${stripshared}");
+#genSO("${root}/lib/libc.so.0", "${uclibc}/mipsel-brcm-linux-uclibc/sysroot/usr/lib/libc.a", "", "-Wl,-init=__uClibc_init ${uclibc}/lib/optinfo/interp.os");
+genSO("${root}/lib/libresolv.so.0", "${uclibc}/mipsel-brcm-linux-uclibc/sysroot/usr/lib/libresolv.a", "${stripshared}");
+genSO("${root}/lib/libcrypt.so.0", "${uclibc}/mipsel-brcm-linux-uclibc/sysroot/usr/lib/libcrypt.a", "${stripshared}");
+genSO("${root}/lib/libm.so.0", "${uclibc}/mipsel-brcm-linux-uclibc/sysroot/usr/lib/libm.a");
+genSO("${root}/lib/libpthread.so.0", "${uclibc}/mipsel-brcm-linux-uclibc/sysroot/usr/lib/libpthread.a", "${stripshared}", "-u pthread_mutexattr_init -Wl,-init=__pthread_initialize_minimal_internal");
+genSO("${root}/lib/libutil.so.0", "${uclibc}/mipsel-brcm-linux-uclibc/sysroot/usr/lib/libutil.a", "${stripshared}");
+genSO("${root}/lib/libdl.so.0", "${uclibc}/mipsel-brcm-linux-uclibc/sysroot/usr/lib/libdl.a", "${stripshared}");
+genSO("${root}/lib/libnsl.so.0", "${uclibc}/mipsel-brcm-linux-uclibc/sysroot/usr/lib/libnsl.a", "${stripshared}");
 
 if ($openssldir eq "openssl") {
-	genSO("${root}/usr/lib/libcrypto.so.1.0.0", "${router}/${openssldir}/libcrypto.a");
-	genSO("${root}/usr/lib/libssl.so.1.0.0", "${router}/${openssldir}/libssl.a", "${stripshared}", "-L${router}/${openssldir}");
+	genSO("${root}/usr/lib/libcrypto.so.1.0.0", "${openssldir}/libcrypto.so.1.0.0");
+	genSO("${root}/usr/lib/libssl.so.1.0.0", "${openssldir}/libssl.so.1.0.0", "${stripshared}", "-L${router}/${openssldir}");
 }
 else {
-	genSO("${root}/usr/lib/libcrypto.so.1.1", "${router}/${openssldir}/libcrypto.a");
-#	genSO("${root}/usr/lib/libssl.so.1.1", "${router}/${openssldir}/libssl.a", "${stripshared}", "-L${router}/${openssldir}");
+	genSO("${root}/usr/lib/libcrypto.so.1.1", "${openssldir}/libcrypto.so.1.1");
+	genSO("${root}/usr/lib/libssl.so.1.1", "${openssldir}/libssl.so.1.1", "${stripshared}", "-L${router}/${openssldir}");
 }
 
 genSO("${root}/usr/lib/libzebra.so", "${router}/zebra/lib/libzebra.a");
 genSO("${root}/usr/lib/libz.so.1", "${router}/zlib/libz.a");
 genSO("${root}/usr/lib/libjpeg.so", "${router}/jpeg/libjpeg.a");
-genSO("${root}/usr/lib/libsqlite3.so.0.8.6", "${router}/sqlite/.libs/libsqlite3.a");
+genSO("${root}/usr/lib/libsqlite3.so.0.8.6", "${router}/sqlite/.libs/libsqlite3.so.0.8.6");
 genSO("${root}/usr/lib/libvorbis.so.0", "${router}/libvorbis/lib/.libs/libvorbis.a", "", "-L${router}/libogg/src/.libs");
 genSO("${root}/usr/lib/libid3tag.so.0", "${router}/libid3tag/.libs/libid3tag.a", "", "-L${router}/zlib");
 genSO("${root}/usr/lib/libexif.so.12", "${router}/libexif/libexif/.libs/libexif.a");
@@ -616,39 +621,38 @@
 genSO("${root}/usr/lib/libavformat.so.52", "${router}/ffmpeg/libavformat/libavformat.a", "", "-L${router}/ffmpeg/libavutil -L${router}/ffmpeg/libavcodec -L${router}/zlib");
 
 genSO("${root}/usr/lib/liblzo2.so.2.0.0", "${router}/lzo/src/.libs/liblzo2.a");
-#genSO("${root}/usr/lib/libiptc.so", "${router}/iptables/libiptc/libiptc.a");
-#genSO("${root}/usr/lib/libshared.so", "${router}/shared/libshared.a");
-#genSO("${root}/usr/lib/libnvram.so", "${router}/nvram/libnvram.a");
-#genSO("${root}/usr/lib/libusb-1.0.so.0", "${router}/libusb10/libusb/.libs/libusb-1.0.a");
-#genSO("${root}/usr/lib/libusb-0.1.so.4", "${router}/libusb/libusb/.libs/libusb.a", "", "-L${router}/libusb10/libusb/.libs");
+genSO("${root}/usr/lib/libiptc.so", "${router}/iptables/libiptc/libiptc.a");
+genSO("${root}/usr/lib/libshared.so", "${router}/shared/libshared.a");
+genSO("${root}/usr/lib/libnvram.so", "${router}/nvram/libnvram.a");
+genSO("${root}/usr/lib/libusb-1.0.so.0", "${router}/libusb10/libusb/.libs/libusb-1.0.a");
+genSO("${root}/usr/lib/libusb-0.1.so.4", "${router}/libusb/libusb/.libs/libusb.a", "", "-L${router}/libusb10/libusb/.libs");
 
 genSO("${root}/usr/lib/libbcmcrypto.so", "${router}/libbcmcrypto/libbcmcrypto.a");
 
 #shibby
-genSO("${root}/usr/lib/libcurl.so.4.7.0", "${router}/libcurl/lib/.libs/libcurl.a", "", "-L${router}/zlib -L${router}/${openssldir}");
-genSO("${root}/usr/lib/libevent-2.1.so.7", "${router}/libevent/.libs/libevent.a");
-genSO("${root}/usr/lib/libdaemon.so.0.5.0", "${router}/libdaemon/libdaemon/.libs/libdaemon.a");
+#genSO("${root}/usr/lib/libcurl.so.4.7.0", "${router}/libcurl/lib/.libs/libcurl.a", "", "-L${router}/zlib -L${router}/${openssldir}");
+genSO("${root}/usr/lib/libevent-2.1.so.7", "${router}/libevent/.libs/libevent.a", "", "-L${router}/${openssldir}");
+genSO("${root}/usr/lib/libdaemon.so.0.5.0", "${router}/libdaemon/libdaemon/.libs/libdaemon.so.0.5.0");
 genSO("${root}/usr/lib/libiconv.so.2.6.1", "${router}/libiconv/lib/.libs/libiconv.a");
 genSO("${root}/usr/lib/libnfnetlink.so.0.2.0", "${router}/libnfnetlink/src/.libs/libnfnetlink.a");
 genSO("${root}/usr/lib/libsodium.so.23.3.0", "${router}/libsodium/src/libsodium/.libs/libsodium.a");
-genSO("${root}/usr/lib/libpng16.so.16.37.0", "${router}/libpng/.libs/libpng16.a", "", "-L${router}/zlib");
+genSO("${root}/usr/lib/libpng16.so.16.37.0", "${router}/libpng/.libs/libpng16.so.16.37.0", "", "-L${router}/zlib");
 genSO("${root}/usr/lib/libxml2.so.2.9.12", "${router}/libxml2/.libs/libxml2.a", "", "-L${router}/zlib");
-genSO("${root}/usr/lib/libpcre.so.1.2.12", "${router}/pcre/.libs/libpcre.a");
+genSO("${root}/usr/lib/libpcre.so.1.2.13", "${router}/pcre/.libs/libpcre.so.1.2.12");
 genSO("${root}/usr/lib/libpcreposix.so.0.0.7", "${router}/pcre/.libs/libpcreposix.a");
 genSO("${root}/usr/lib/libatomic_ops.so.1.0.3", "${router}/libatomic_ops/src/.libs/libatomic_ops.a");
 genSO("${root}/usr/lib/libncurses.so.6", "${router}/libncurses/lib/libncurses.a");
 
 # iperf (pedro)
-genSO("${root}/usr/lib/libiperf.so.0.0.0", "${router}/iperf/src/.libs/libiperf.a");
+genSO("${root}/usr/lib/libiperf.so.0.0.0", "${router}/iperf/src/.libs/libiperf.so.0.0.0");
 
 # e2fsprogs (pedro)
-genSO("${root}/usr/lib/libext2fs.so.2.4", "${router}/e2fsprogs/lib/libext2fs.a", "", "-L${router}/e2fsprogs/lib");
-genSO("${root}/usr/lib/libuuid.so.1.2", "${router}/e2fsprogs/lib/libuuid.a", "", "-L${router}/e2fsprogs/lib");
-genSO("${root}/usr/lib/libblkid.so.1.0", "${router}/e2fsprogs/lib/libblkid.a", "", "-L${router}/e2fsprogs/lib");
-genSO("${root}/usr/lib/libe2p.so.2.3", "${router}/e2fsprogs/lib/libe2p.a", "", "-L${router}/e2fsprogs/lib");
-genSO("${root}/usr/lib/libcom_err.so.2.1", "${router}/e2fsprogs/lib/libcom_err.a", "", "-L${router}/e2fsprogs/lib");
-
-print "\n--- end ---\n\n";
+genSO("${root}/usr/lib/libext2fs.so.2.4", "${router}/e2fsprogs/lib/libext2fs.so.2.4", "", "-L${router}/e2fsprogs/lib");
+genSO("${root}/usr/lib/libuuid.so.1.2", "${router}/e2fsprogs/lib/libuuid.so.1.2", "", "-L${router}/e2fsprogs/lib");
+genSO("${root}/usr/lib/libblkid.so.1.0", "${router}/e2fsprogs/lib/libblkid.so.1.0","", "-L${router}/e2fsprogs/lib");
+genSO("${root}/usr/lib/libe2p.so.2.3", "${router}/e2fsprogs/lib/libe2p.so", "", "-L${router}/e2fsprogs/lib");
+genSO("${root}/usr/lib/libcom_err.so.2.1", "${router}/e2fsprogs/lib/libcom_err.so", "", "-L${router}/e2fsprogs/lib");
+print "\n";
 
 close(LOG);
 exit(0);
