--- a/release/src/router/iptables/libiptc/libip6tc.c	2022-01-24 14:05:37.476249900 +0100
+++ b/gcc-5.3-toolchain_mips/libip6tc.c	2022-01-04 11:43:13.000000000 +0100
@@ -113,7 +113,7 @@
 #include "libiptc.c"
 
 #define BIT6(a, l) \
- ((ntohl(a->in6_u.u6_addr32[(l) / 32]) >> (31 - ((l) & 31))) & 1)
+ ((ntohl(a->s6_addr32[(l) / 32]) >> (31 - ((l) & 31))) & 1)
 
 int
 ipv6_prefix_length(const struct in6_addr *a)
