--- a/release/src/router/samba3/Makefile	2022-01-24 14:05:37.476249900 +0100
+++ b/gcc-5.3-toolchain_mips/Makefile_samba	2022-01-03 12:19:55.000000000 +0100
@@ -5,8 +5,8 @@
 
 export PARALLEL_BUILD := -j`grep -c '^processor' /proc/cpuinfo`
 
-SMBCFLAGS = $(EXTRACFLAGS) $(CFLAG_OPTIMIZE) -ffunction-sections -fdata-sections $(OPTSIZE_FLAG) $(if $(or $(TCONFIG_BCMARM),$(TCONFIG_NGINX)),-I$(TOP)/libiconv/staged/usr/include,)
-SMBLDFLAGS = -ffunction-sections -fdata-sections -Wl,--gc-sections $(if $(or $(TCONFIG_BCMARM),$(TCONFIG_NGINX)),-L$(TOP)/libiconv/staged/usr/lib,)
+SMBCFLAGS = $(EXTRACFLAGS) $(CFLAG_OPTIMIZE) -ffunction-sections -fdata-sections -I$(TOP)/zlib/staged/usr/include $(OPTSIZE_FLAG) $(if $(or $(TCONFIG_BCMARM),$(TCONFIG_NGINX)),-I$(TOP)/libiconv/staged/usr/include,)
+SMBLDFLAGS = -ffunction-sections -fdata-sections -Wl,--gc-sections -L$(TOP)/zlib/staged/usr/lib -lz $(if $(or $(TCONFIG_BCMARM),$(TCONFIG_NGINX)),-L$(TOP)/libiconv/staged/usr/lib,)
 
 SMB_USE_MULTICALL=1
 
