--- a/release/src-rt-6.x/linux/linux-2.6/Makefile	2022-01-24 14:05:37.466249900 +0100
+++ b/gcc-5.3-toolchain_mips/Makefile_linux	2022-01-24 12:34:23.896249900 +0100
@@ -194,7 +194,7 @@
 
 HOSTCC       = gcc
 HOSTCXX      = g++
-HOSTCFLAGS   = -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer
+HOSTCFLAGS   = -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer -std=gnu89
 HOSTCXXFLAGS = -O2
 
 # Decide whether to build built-in, modular, or both.
@@ -310,7 +310,7 @@
 CPPFLAGS        := -D__KERNEL__ $(LINUXINCLUDE)
 
 CFLAGS          := -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs \
-                   -fno-strict-aliasing -fno-common
+                   -fno-strict-aliasing -fno-common -Wno-unused
 AFLAGS          := -D__ASSEMBLY__
 
 # Broadcom source tree
