--- a/release/src/router/hotplug2/mem_utils.c	2022-01-24 14:05:37.476249900 +0100
+++ b/gcc-5.3-toolchain_mips/mem_utils.c	2021-12-31 12:30:09.000000000 +0100
@@ -16,7 +16,7 @@
  *
  * Returns: Pointer to freshly allocated memory
  */
-inline void *xmalloc(size_t size) {
+extern inline void *xmalloc(size_t size) {
 	void *ptr;
 	ptr = malloc(size);
 	if (ptr == NULL) {
@@ -34,7 +34,7 @@
  *
  * Returns: Pointer to reallocated memory
  */
-inline void *xrealloc(void *inptr, size_t size) {
+extern inline void *xrealloc(void *inptr, size_t size) {
 	void *ptr;
 	ptr = realloc(inptr, size);
 	if (ptr == NULL) {
