--- a/release/src/router/rc/services.c	2022-01-24 14:05:37.476249900 +0100
+++ b/gcc-5.3-toolchain_mips/services.c	2021-12-31 12:30:09.000000000 +0100
@@ -2965,7 +2995,7 @@
 	unsigned char ea[ETHER_ADDR_LEN];
 	char serial[18], uuid[37];
 	char *buf, *p, *q;
-	char *path, *restrict;
+	char *path, *restrct;
 
 	if (!nvram_get_int("ms_enable"))
 		return;
@@ -3034,11 +3064,11 @@
 				/* path<restrict[A|V|P|] */
 				p = buf;
 				while ((q = strsep(&p, ">")) != NULL) {
-					if ((vstrsep(q, "<", &path, &restrict) < 1) || (!path) || (!*path))
+					if ((vstrsep(q, "<", &path, &restrct) < 1) || (!path) || (!*path))
 						continue;
 
 					fprintf(f, "media_dir=%s%s%s\n",
-						restrict ? : "", (restrict && *restrict) ? "," : "", path);
+						restrct ? : "", (restrct && *restrct) ? "," : "", path);
 				}
 				free(buf);
 			}
