--- a/release/src/router/shared/shutils.h	2022-01-24 14:05:37.496249900 +0100
+++ b/gcc-5.3-toolchain_mips/shutils.h	2021-12-31 12:30:09.000000000 +0100
@@ -137,7 +137,7 @@
 
 /* Check for a blank character; that is, a space or a tab */
 #ifndef isblank
-#define isblank(c) ((c) == ' ' || (c) == '\t')
+#define isblank(c) (c == ' ' || c == '\t')
 #endif
 
 /* Strip trailing CR/NL from string <s> */
