*** Makefile_a	2019-11-23 19:17:57.383534020 +0100
--- Makefile_b	2019-11-03 17:23:12.589942849 +0100
***************
*** 38,44 ****
  #
  #
  SEP=`$(eval progress=$(shell echo $$(($(progress)+1))))` \
! echo "\033[41;1m   $@ ${progress}$\/${totalSteps} \033[0m\033]2;Building $@ ${progress}$\/${totalSteps}\007"
  
  
  ifeq ($(TCONFIG_OPTIMIZE_SIZE),y)
--- 38,44 ----
  #
  #
  SEP=`$(eval progress=$(shell echo $$(($(progress)+1))))` \
! printf "\033[41;1m   $@ ${progress}$\/${totalSteps} \033[0m\033]2;Building $@ ${progress}$\/${totalSteps}\007"
  
  
  ifeq ($(TCONFIG_OPTIMIZE_SIZE),y)
***************
*** 319,325 ****
  	@echo ${totalSteps}
  
  install package: countInstallSteps $(obj-install) $(LINUXDIR)/.config
! 	@echo "\033[41;1m   Installing \033[0m\033]2;Installing\007"
  	install -d $(TARGETDIR)
  
  
--- 319,325 ----
  	@echo ${totalSteps}
  
  install package: countInstallSteps $(obj-install) $(LINUXDIR)/.config
! 	@printf "\033[41;1m   Installing \033[0m\033]2;Installing\007"
  	install -d $(TARGETDIR)
***************
*** 1880,1886 ****
  
  php/stamp-h1: pcre zlib libiconv sqlite libxml2 libpng jpeg libcurl
  	$(call patch_files,php)
! 	cd php && CC=$(CC) CXX=$(ARCH)-g++ AR=$(AR) RANLIB=$(RANLIB) LD=$(LD) \
  	CFLAGS="-Os -Wall -I$(TOP)/zlib -I$(TOP)/libxml2/include/libxml -I$(TOP)/libxml2/include -I$(TOP)/pcre -I$(TOP)/libiconv/include \
  		-I$(TOP)/libpng/staged/usr/include -I$(TOP)/libcurl/staged/usr/include" \
  	LDFLAGS="-L$(TOP)/pcre/.libs -L$(TOP)/sqlite/.libs -L$(TOP)/zlib -L$(TOP)/libxml2/.libs -L$(TOP)/libiconv/lib/.libs \
--- 1880,1886 ----
  
  php/stamp-h1: pcre zlib libiconv sqlite libxml2 libpng jpeg libcurl
  	$(call patch_files,php)
! 	cd php && CC=$(CC) CXX=$(CROSS_COMPILE)g++ AR=$(AR) RANLIB=$(RANLIB) LD=$(LD) \
  	CFLAGS="-Os -Wall -I$(TOP)/zlib -I$(TOP)/libxml2/include/libxml -I$(TOP)/libxml2/include -I$(TOP)/pcre -I$(TOP)/libiconv/include \
  		-I$(TOP)/libpng/staged/usr/include -I$(TOP)/libcurl/staged/usr/include" \
  	LDFLAGS="-L$(TOP)/pcre/.libs -L$(TOP)/sqlite/.libs -L$(TOP)/zlib -L$(TOP)/libxml2/.libs -L$(TOP)/libiconv/lib/.libs \
***************
*** 2365,2372 ****
  
  portmap/stamp-h1:
  	$(call patch_files,portmap)
! 	cd portmap \
! 	$(MAKE) -C portmap CC=$(CC) AR=$(AR) LD=$(LD) RANLIB=$(RANLIB)
  	touch portmap/stamp-h1
  
  portmap: portmap/stamp-h1
--- 2365,2371 ----
  
  portmap/stamp-h1:
  	$(call patch_files,portmap)
! 	cd portmap
  	touch portmap/stamp-h1
  
  portmap: portmap/stamp-h1
***************
*** 2471,2477 ****
  
  sd-idle/stamp-h1:
  	@$(SEP)
- 	cd sd-idle \
  	CFLAGS="-Os -Wall --host=$(HOST) --target=$(HOST) $(EXTRACFLAGS)" \
  	$(MAKE) -C sd-idle
  	chmod 0755 sd-idle/sd-idle
--- 2470,2475 ----
