--- services.c.orig	2022-06-11 15:43:23.000000000 +0000
+++ services.c	2022-06-11 15:43:27.000000000 +0000
@@ -3041,7 +3041,7 @@
 	unsigned char ea[ETHER_ADDR_LEN];
 	char serial[18], uuid[37];
 	char *buf, *p, *q;
-	char *path, *restrict;
+	char *path, *restrct;
 
 	if (!nvram_get_int("ms_enable"))
 		return;
@@ -3110,11 +3110,11 @@
 				/* path<restrict[A|V|P|] */
 				p = buf;
 				while ((q = strsep(&p, ">")) != NULL) {
-					if ((vstrsep(q, "<", &path, &restrict) < 1) || (!path) || (!*path))
+					if ((vstrsep(q, "<", &path, &restrct) < 1) || (!path) || (!*path))
 						continue;
 
 					fprintf(f, "media_dir=%s%s%s\n",
-						restrict ? : "", (restrict && *restrict) ? "," : "", path);
+						restrct ? : "", (restrct && *restrct) ? "," : "", path);
 				}
 				free(buf);
 			}
