*** /home/stephan/freshtomato-arm/release/src-rt-6.x.4708/router/miniupnpd/Makefile.linux	2019-11-02 17:20:00.019902507 +0100
--- Makefile.linux	2019-10-12 15:30:34.379976000 +0200
***************
*** 170,176 ****
  
  TEST := $(shell $(PKG_CONFIG) --exists uuid && echo 1)
  ifeq ($(TEST),1)
! LDLIBS += $(shell $(PKG_CONFIG) --static --libs-only-l uuid)
  else
  $(info please install uuid-dev package / libuuid)
  endif # ($(TEST),1)
--- 170,176 ----
  
  TEST := $(shell $(PKG_CONFIG) --exists uuid && echo 1)
  ifeq ($(TEST),1)
! # LDLIBS += $(shell $(PKG_CONFIG) --static --libs-only-l uuid)
  else
  $(info please install uuid-dev package / libuuid)
  endif # ($(TEST),1)
