*** Makefile	2020-01-25 19:20:37.620373000 +0100
--- Makefile_mod	2020-01-25 19:25:12.543710926 +0100
***************
*** 38,44 ****
  #
  #
  SEP=`$(eval progress=$(shell echo $$(($(progress)+1))))` \
! echo "\033[41;1m   $@ ${progress}$\/${totalSteps} \033[0m\033]2;Building $@ ${progress}$\/${totalSteps}\007"
  
  export CFLAGS += -DBCMWPA2
  ifeq ($(TCONFIG_BCMWL6),y)
--- 38,44 ----
  #
  #
  SEP=`$(eval progress=$(shell echo $$(($(progress)+1))))` \
! printf "\033[41;1m   $@ ${progress}$\/${totalSteps} \033[0m\033]2;Building $@ ${progress}$\/${totalSteps}\007"
  
  export CFLAGS += -DBCMWPA2
  ifeq ($(TCONFIG_BCMWL6),y)
***************
*** 57,62 ****
--- 57,64 ----
  #
  # standard packages
  #
+ 
+ obj-y += tor
  ifeq ($(TCONFIG_BCMARM),y)
  obj-y += libbcm
  else
***************
*** 128,134 ****
  obj-y += e2fsprogs
  endif
  
! obj-$(TCONFIG_TOR) += tor
  obj-$(TCONFIG_DNSCRYPT) += libsodium
  obj-$(TCONFIG_DNSCRYPT) += dnscrypt
  obj-$(TCONFIG_STUBBY) += libyaml
--- 130,136 ----
  obj-y += e2fsprogs
  endif
  
! #obj-$(TCONFIG_TOR) += tor
  obj-$(TCONFIG_DNSCRYPT) += libsodium
  obj-$(TCONFIG_DNSCRYPT) += dnscrypt
  obj-$(TCONFIG_STUBBY) += libyaml
***************
*** 416,422 ****
  	@echo ${totalSteps}
  
  install package: countInstallSteps $(obj-install) $(LINUXDIR)/.config
! 	@echo "\033[41;1m   Installing \033[0m\033]2;Installing\007"
  	install -d $(TARGETDIR)
  
  
--- 418,424 ----
  	@echo ${totalSteps}
  
  install package: countInstallSteps $(obj-install) $(LINUXDIR)/.config
! 	@printf "\033[41;1m   Installing \033[0m\033]2;Installing\007"
  	install -d $(TARGETDIR)
  
  
***************
*** 1564,1570 ****
  	cd $(TOP)/flac && autoreconf -fsi && \
  	CFLAGS="-Os $(EXTRACFLAGS) -fPIC -ffunction-sections -fdata-sections" \
  	CPPFLAGS="-I$(TOP)/libogg/include -I$(LINUXDIR)/include" \
! 	LDFLAGS="-L$(TOP)/libogg/src/.libs -fPIC -ffunction-sections -fdata-sections -Wl,--gc-sections" \
  	$(CONFIGURE) --enable-shared --enable-static --prefix='' --disable-rpath \
  		--disable-doxygen-docs --disable-xmms-plugin --disable-cpplibs --disable-thorough-tests \
  		--without-libiconv-prefix --disable-altivec --disable-sse --disable-dependency-tracking
--- 1566,1572 ----
  	cd $(TOP)/flac && autoreconf -fsi && \
  	CFLAGS="-Os $(EXTRACFLAGS) -fPIC -ffunction-sections -fdata-sections" \
  	CPPFLAGS="-I$(TOP)/libogg/include -I$(LINUXDIR)/include" \
! 	LDFLAGS="-L$(TOP)/libogg/src/.libs -L$(TOOLCHAIN)/lib -fPIC -ffunction-sections -fdata-sections -Wl,--gc-sections" \
  	$(CONFIGURE) --enable-shared --enable-static --prefix='' --disable-rpath \
  		--disable-doxygen-docs --disable-xmms-plugin --disable-cpplibs --disable-thorough-tests \
  		--without-libiconv-prefix --disable-altivec --disable-sse --disable-dependency-tracking
***************
*** 1944,1950 ****
  	cd libxml2 && \
  	autoreconf -fsi && \
  	CC=$(CC) AR=$(AR) RANLIB=$(RANLIB) LD=$(LD) CFLAGS="-Os -Wall $(EXTRACFLAGS)" \
! 	$(CONFIGURE) --prefix=/usr --without-python --enable-static --enable-shared \
  		--includedir="$(TOP)/libxml2/include" --without-lzma --disable-dependency-tracking
  	touch $@
  
--- 1946,1952 ----
  	cd libxml2 && \
  	autoreconf -fsi && \
  	CC=$(CC) AR=$(AR) RANLIB=$(RANLIB) LD=$(LD) CFLAGS="-Os -Wall $(EXTRACFLAGS)" \
! 	$(CONFIGURE) --prefix=/usr --without-python --enable-static --enable-shared --with-zlib="$(TOP)/zlib/staged/usr" \
  		--includedir="$(TOP)/libxml2/include" --without-lzma --disable-dependency-tracking
  	touch $@
  
***************
*** 2001,2007 ****
  
  php/stamp-h1: pcre zlib libiconv sqlite libxml2 libpng jpeg libcurl
  	$(call patch_files,php)
! 	cd php && CC=$(CC) CXX=$(ARCH)-g++ AR=$(AR) RANLIB=$(RANLIB) LD=$(LD) \
  	CFLAGS="-Os -Wall -I$(TOP)/zlib -I$(TOP)/libxml2/include/libxml -I$(TOP)/libxml2/include -I$(TOP)/pcre -I$(TOP)/libiconv/include \
  		-I$(TOP)/libpng/staged/usr/include -I$(TOP)/libcurl/staged/usr/include" \
  	LDFLAGS="-L$(TOP)/pcre/.libs -L$(TOP)/sqlite/.libs -L$(TOP)/zlib -L$(TOP)/libxml2/.libs -L$(TOP)/libiconv/lib/.libs \
--- 2003,2009 ----
  
  php/stamp-h1: pcre zlib libiconv sqlite libxml2 libpng jpeg libcurl
  	$(call patch_files,php)
! 	cd php && CC=$(CC) CXX=$(HOST)-g++ AR=$(AR) RANLIB=$(RANLIB) LD=$(LD) \
  	CFLAGS="-Os -Wall -I$(TOP)/zlib -I$(TOP)/libxml2/include/libxml -I$(TOP)/libxml2/include -I$(TOP)/pcre -I$(TOP)/libiconv/include \
  		-I$(TOP)/libpng/staged/usr/include -I$(TOP)/libcurl/staged/usr/include" \
  	LDFLAGS="-L$(TOP)/pcre/.libs -L$(TOP)/sqlite/.libs -L$(TOP)/zlib -L$(TOP)/libxml2/.libs -L$(TOP)/libiconv/lib/.libs \
***************
*** 2478,2485 ****
  
  portmap/stamp-h1:
  	$(call patch_files,portmap)
! 	cd portmap \
! 	$(MAKE) -C portmap
  	@touch $@
  
  portmap: portmap/stamp-h1
--- 2480,2486 ----
  
  portmap/stamp-h1:
  	$(call patch_files,portmap)
! 	cd portmap
  	@touch $@
  
  portmap: portmap/stamp-h1
***************
*** 2593,2599 ****
  
  sd-idle/stamp-h1:
  	@$(SEP)
- 	cd sd-idle \
  	CFLAGS="-Os -Wall --host=$(HOST) --target=$(HOST) $(EXTRACFLAGS)" \
  	$(MAKE) -C sd-idle
  	chmod 0755 sd-idle/sd-idle
--- 2594,2599 ----
***************
*** 2814,2819 ****
--- 2814,2820 ----
  	@$(SEP)
  	$(MAKE) -C libmnl
  	$(MAKE) -C libmnl DESTDIR=$(TOP)/libmnl/staged install
+ 	@rm -f libmnl/staged/usr/lib/libmnl.la
  
  libmnl-install:
  	install -d $(INSTALLDIR)/libmnl/usr/lib/
***************
*** 2830,2839 ****
  
  ipset/stamp-h1: libmnl
  	$(call patch_files,ipset)
! 	cd $(TOP)/ipset && CC=$(CC) STRIP=$(STRIP) \
! 	autoreconf -fsi && \
  	CFLAGS="-Os -Wall $(EXTRACFLAGS) -ffunction-sections -fdata-sections" \
! 	LDFLAGS="-ffunction-sections -fdata-sections -Wl,--gc-sections" \
  	libmnl_CFLAGS="-I$(TOP)/libmnl/staged/usr/include" \
  	libmnl_LIBS="-L$(TOP)/libmnl/staged/usr/lib -lmnl" \
  	$(CONFIGURE) --prefix=/usr --with-kmod=no
--- 2831,2839 ----
  
  ipset/stamp-h1: libmnl
  	$(call patch_files,ipset)
! 	cd $(TOP)/ipset && autoreconf -fsi && CC=$(CC) STRIP=$(STRIP) \
  	CFLAGS="-Os -Wall $(EXTRACFLAGS) -ffunction-sections -fdata-sections" \
! 	LDFLAGS="-ffunction-sections -fdata-sections -Wl,--gc-sections -L$(TOP)/libmnl/staged/usr/lib -lmnl" \
  	libmnl_CFLAGS="-I$(TOP)/libmnl/staged/usr/include" \
  	libmnl_LIBS="-L$(TOP)/libmnl/staged/usr/lib -lmnl" \
  	$(CONFIGURE) --prefix=/usr --with-kmod=no
