*** Makefile	2020-01-25 19:20:37.620373000 +0100
--- Makefile_mod	2020-01-25 19:25:12.543710926 +0100
***************
*** 38,44 ****
  #
  #
  SEP=`$(eval progress=$(shell echo $$(($(progress)+1))))` \
! echo "\033[41;1m   $@ ${progress}$\/${totalSteps} \033[0m\033]2;Building $@ ${progress}$\/${totalSteps}\007"
  
  export CFLAGS += -DBCMWPA2
  ifeq ($(TCONFIG_BCMWL6),y)
--- 38,44 ----
  #
  #
  SEP=`$(eval progress=$(shell echo $$(($(progress)+1))))` \
! printf "\033[41;1m   $@ ${progress}$\/${totalSteps} \033[0m\033]2;Building $@ ${progress}$\/${totalSteps}\007"
  
  export CFLAGS += -DBCMWPA2
  ifeq ($(TCONFIG_BCMWL6),y)
***************
*** 416,422 ****
  	@echo ${totalSteps}
  
  install package: countInstallSteps $(obj-install) $(LINUXDIR)/.config
! 	@echo "\033[41;1m   Installing \033[0m\033]2;Installing\007"
  	install -d $(TARGETDIR)
  
  
--- 418,424 ----
  	@echo ${totalSteps}
  
  install package: countInstallSteps $(obj-install) $(LINUXDIR)/.config
! 	@printf "\033[41;1m   Installing \033[0m\033]2;Installing\007"
  	install -d $(TARGETDIR)
  
  
***************
*** 1564,1570 ****
  	cd $(TOP)/flac && autoreconf -fsi && \
  	CFLAGS="-Os $(EXTRACFLAGS) -fPIC -ffunction-sections -fdata-sections" \
  	CPPFLAGS="-I$(TOP)/libogg/include -I$(LINUXDIR)/include" \
! 	LDFLAGS="-L$(TOP)/libogg/src/.libs -fPIC -ffunction-sections -fdata-sections -Wl,--gc-sections" \
  	$(CONFIGURE) --enable-shared --enable-static --prefix='' --disable-rpath \
  		--disable-doxygen-docs --disable-xmms-plugin --disable-cpplibs --disable-thorough-tests \
  		--without-libiconv-prefix --disable-altivec --disable-sse --disable-dependency-tracking
--- 1566,1572 ----
  	cd $(TOP)/flac && autoreconf -fsi && \
  	CFLAGS="-Os $(EXTRACFLAGS) -fPIC -ffunction-sections -fdata-sections" \
  	CPPFLAGS="-I$(TOP)/libogg/include -I$(LINUXDIR)/include" \
! 	LDFLAGS="-L$(TOP)/libogg/src/.libs -L$(TOOLCHAIN)/lib -fPIC -ffunction-sections -fdata-sections -Wl,--gc-sections" \
  	$(CONFIGURE) --enable-shared --enable-static --prefix='' --disable-rpath \
  		--disable-doxygen-docs --disable-xmms-plugin --disable-cpplibs --disable-thorough-tests \
  		--without-libiconv-prefix --disable-altivec --disable-sse --disable-dependency-tracking
***************
*** 2001,2007 ****
  
  php/stamp-h1: pcre zlib libiconv sqlite libxml2 libpng jpeg libcurl
  	$(call patch_files,php)
! 	cd php && CC=$(CC) CXX=$(ARCH)-g++ AR=$(AR) RANLIB=$(RANLIB) LD=$(LD) \
  	CFLAGS="-Os -Wall -I$(TOP)/zlib -I$(TOP)/libxml2/include/libxml -I$(TOP)/libxml2/include -I$(TOP)/pcre -I$(TOP)/libiconv/include \
  		-I$(TOP)/libpng/staged/usr/include -I$(TOP)/libcurl/staged/usr/include" \
  	LDFLAGS="-L$(TOP)/pcre/.libs -L$(TOP)/sqlite/.libs -L$(TOP)/zlib -L$(TOP)/libxml2/.libs -L$(TOP)/libiconv/lib/.libs \
--- 2003,2009 ----
  
  php/stamp-h1: pcre zlib libiconv sqlite libxml2 libpng jpeg libcurl
  	$(call patch_files,php)
! 	cd php && CC=$(CC) CXX=$(CROSS-COMPILE)g++ AR=$(AR) RANLIB=$(RANLIB) LD=$(LD) \
  	CFLAGS="-Os -Wall -I$(TOP)/zlib -I$(TOP)/libxml2/include/libxml -I$(TOP)/libxml2/include -I$(TOP)/pcre -I$(TOP)/libiconv/include \
  		-I$(TOP)/libpng/staged/usr/include -I$(TOP)/libcurl/staged/usr/include" \
  	LDFLAGS="-L$(TOP)/pcre/.libs -L$(TOP)/sqlite/.libs -L$(TOP)/zlib -L$(TOP)/libxml2/.libs -L$(TOP)/libiconv/lib/.libs \
***************
*** 2478,2485 ****
  
  portmap/stamp-h1:
  	$(call patch_files,portmap)
! 	cd portmap \
! 	$(MAKE) -C portmap
  	@touch $@
  
  portmap: portmap/stamp-h1
--- 2480,2486 ----
  
  portmap/stamp-h1:
  	$(call patch_files,portmap)
! 	cd portmap
  	@touch $@
  
  portmap: portmap/stamp-h1
***************
*** 2593,2599 ****
  
  sd-idle/stamp-h1:
  	@$(SEP)
- 	cd sd-idle \
  	CFLAGS="-Os -Wall --host=$(HOST) --target=$(HOST) $(EXTRACFLAGS)" \
  	$(MAKE) -C sd-idle
  	chmod 0755 sd-idle/sd-idle
--- 2594,2599 ----
***************
*** 2814,2819 ****
--- 2814,2820 ----
  	@$(SEP)
  	$(MAKE) -C libmnl
  	$(MAKE) -C libmnl DESTDIR=$(TOP)/libmnl/staged install
+ 	@rm -f libmnl/staged/usr/lib/libmnl.la
  
  libmnl-install:
  	install -d $(INSTALLDIR)/libmnl/usr/lib/
