*** /home/stephan/freshtomato-arm/release/src-rt-6.x.4708/router/Makefile	2019-11-23 17:32:46.893347776 +0100
--- Makefile	2019-11-23 17:30:28.290010404 +0100
***************
*** 38,44 ****
  #
  #
  SEP=`$(eval progress=$(shell echo $$(($(progress)+1))))` \
! echo "\033[41;1m   $@ ${progress}$\/${totalSteps} \033[0m\033]2;Building $@ ${progress}$\/${totalSteps}\007"
  
  export CFLAGS += -DBCMWPA2
  ifeq ($(TCONFIG_BCMWL6),y)
--- 38,44 ----
  #
  #
  SEP=`$(eval progress=$(shell echo $$(($(progress)+1))))` \
! printf "\033[41;1m   $@ ${progress}$\/${totalSteps} \033[0m\033]2;Building $@ ${progress}$\/${totalSteps}\007"
  
  export CFLAGS += -DBCMWPA2
  ifeq ($(TCONFIG_BCMWL6),y)
***************
*** 1633,1639 ****
  	cd $(TOP)/flac && autoreconf -fsi && \
  	CFLAGS="-Os $(EXTRACFLAGS) -fPIC -ffunction-sections -fdata-sections" \
  	CPPFLAGS="-I$(TOP)/libogg/include -I$(LINUXDIR)/include" \
! 	LDFLAGS="-L$(TOP)/libogg/src/.libs -fPIC -ffunction-sections -fdata-sections -Wl,--gc-sections" \
  	$(CONFIGURE) --enable-shared --enable-static --prefix='' --disable-rpath \
  		--disable-doxygen-docs --disable-xmms-plugin --disable-cpplibs --disable-thorough-tests \
  		--without-libiconv-prefix --disable-altivec --disable-sse --disable-dependency-tracking
--- 1633,1639 ----
  	cd $(TOP)/flac && autoreconf -fsi && \
  	CFLAGS="-Os $(EXTRACFLAGS) -fPIC -ffunction-sections -fdata-sections" \
  	CPPFLAGS="-I$(TOP)/libogg/include -I$(LINUXDIR)/include" \
! 	LDFLAGS="-L$(TOP)/libogg/src/.libs -L$(TOOLCHAIN)/lib -fPIC -ffunction-sections -fdata-sections -Wl,--gc-sections" \
  	$(CONFIGURE) --enable-shared --enable-static --prefix='' --disable-rpath \
  		--disable-doxygen-docs --disable-xmms-plugin --disable-cpplibs --disable-thorough-tests \
  		--without-libiconv-prefix --disable-altivec --disable-sse --disable-dependency-tracking
***************
*** 2013,2019 ****
  	cd libxml2 && \
  	autoreconf -fsi && \
  	CC=$(CC) AR=$(AR) RANLIB=$(RANLIB) LD=$(LD) CFLAGS="-Os -Wall $(EXTRACFLAGS)" \
! 	$(CONFIGURE) --prefix=/usr --without-python --enable-static --enable-shared \
  		--includedir="$(TOP)/libxml2/include" --without-lzma --disable-dependency-tracking
  	touch $@
  
--- 2013,2019 ----
  	cd libxml2 && \
  	autoreconf -fsi && \
  	CC=$(CC) AR=$(AR) RANLIB=$(RANLIB) LD=$(LD) CFLAGS="-Os -Wall $(EXTRACFLAGS)" \
! 	$(CONFIGURE) --prefix=/usr --without-python --enable-static --enable-shared --with-zlib="$(TOP)/zlib/staged/usr" \
  		--includedir="$(TOP)/libxml2/include" --without-lzma --disable-dependency-tracking
  	touch $@
  
***************
*** 2064,2070 ****
  
  php/stamp-h1: pcre zlib libiconv sqlite libxml2 libpng jpeg libcurl
  	$(call patch_files,php)
! 	cd php && CC=$(CC) CXX=$(ARCH)-g++ AR=$(AR) RANLIB=$(RANLIB) LD=$(LD) \
  	CFLAGS="-Os -Wall -I$(TOP)/zlib -I$(TOP)/libxml2/include/libxml -I$(TOP)/libxml2/include -I$(TOP)/pcre -I$(TOP)/libiconv/include \
  		-I$(TOP)/libpng/staged/usr/include -I$(TOP)/libcurl/staged/usr/include" \
  	LDFLAGS="-L$(TOP)/pcre/.libs -L$(TOP)/sqlite/.libs -L$(TOP)/zlib -L$(TOP)/libxml2/.libs -L$(TOP)/libiconv/lib/.libs \
--- 2064,2070 ----
  
  php/stamp-h1: pcre zlib libiconv sqlite libxml2 libpng jpeg libcurl
  	$(call patch_files,php)
! 	cd php && CC=$(CC) CXX=$(HOST)-g++ AR=$(AR) RANLIB=$(RANLIB) LD=$(LD) \
  	CFLAGS="-Os -Wall -I$(TOP)/zlib -I$(TOP)/libxml2/include/libxml -I$(TOP)/libxml2/include -I$(TOP)/pcre -I$(TOP)/libiconv/include \
  		-I$(TOP)/libpng/staged/usr/include -I$(TOP)/libcurl/staged/usr/include" \
  	LDFLAGS="-L$(TOP)/pcre/.libs -L$(TOP)/sqlite/.libs -L$(TOP)/zlib -L$(TOP)/libxml2/.libs -L$(TOP)/libiconv/lib/.libs \
***************
*** 2543,2550 ****
  
  portmap/stamp-h1:
  	$(call patch_files,portmap)
! 	cd portmap \
! 	$(MAKE) -C portmap CC=$(CC) AR=$(AR) LD=$(LD) RANLIB=$(RANLIB)
  	touch portmap/stamp-h1
  
  portmap: portmap/stamp-h1
--- 2543,2549 ----
  
  portmap/stamp-h1:
  	$(call patch_files,portmap)
! 	cd portmap
  	touch portmap/stamp-h1
  
  portmap: portmap/stamp-h1
***************
*** 2661,2667 ****
  
  sd-idle/stamp-h1:
  	@$(SEP)
- 	cd sd-idle \
  	CFLAGS="-Os -Wall --host=$(HOST) --target=$(HOST) $(EXTRACFLAGS)" \
  	$(MAKE) -C sd-idle
  	chmod 0755 sd-idle/sd-idle
--- 2660,2665 ----
***************
*** 2877,2882 ****
--- 2875,2881 ----
  	@$(SEP)
  	$(MAKE) -C libmnl
  	$(MAKE) -C libmnl DESTDIR=$(TOP)/libmnl/staged install
+ 	@rm -f libmnl/staged/usr/lib/libmnl.la
  
  libmnl-install:
  	install -d $(INSTALLDIR)/libmnl/usr/lib/
***************
*** 2893,2902 ****
  
  ipset/stamp-h1: libmnl
  	$(call patch_files,ipset)
! 	cd $(TOP)/ipset && CC=$(CC) STRIP=$(STRIP) \
! 	autoreconf -fsi && \
  	CFLAGS="-Os -Wall $(EXTRACFLAGS) -ffunction-sections -fdata-sections" \
! 	LDFLAGS="-ffunction-sections -fdata-sections -Wl,--gc-sections" \
  	libmnl_CFLAGS="-I$(TOP)/libmnl/staged/usr/include" \
  	libmnl_LIBS="-L$(TOP)/libmnl/staged/usr/lib -lmnl" \
  	$(CONFIGURE) --prefix=/usr --with-kmod=no
--- 2892,2900 ----
  
  ipset/stamp-h1: libmnl
  	$(call patch_files,ipset)
! 	cd $(TOP)/ipset && autoreconf -fsi && CC=$(CC) STRIP=$(STRIP) \
  	CFLAGS="-Os -Wall $(EXTRACFLAGS) -ffunction-sections -fdata-sections" \
! 	LDFLAGS="-ffunction-sections -fdata-sections -Wl,--gc-sections -L$(TOP)/libmnl/staged/usr/lib -lmnl" \
  	libmnl_CFLAGS="-I$(TOP)/libmnl/staged/usr/include" \
  	libmnl_LIBS="-L$(TOP)/libmnl/staged/usr/lib -lmnl" \
  	$(CONFIGURE) --prefix=/usr --with-kmod=no
