--- Makefile.orig	2021-11-20 18:43:56.561958977 +0100
+++ Makefile	2021-11-20 18:48:26.386760895 +0100
@@ -1080,16 +1080,17 @@
 		ln -sf xtables-legacy-multi iptables-save && \
 		ln -sf xtables-legacy-multi iptables
 	install -d $(INSTALLDIR)/iptables-1.8.x/usr/lib/xtables
-	install -D iptables-1.8.x/libiptc/.libs/lib*.so $(INSTALLDIR)/iptables-1.8.x/usr/lib/
-	cd $(INSTALLDIR)/iptables-1.8.x/usr/lib && \
-		ln -sf libip4tc.so libip4tc.so.2 && \
-		ln -sf libip4tc.so libip4tc.so.2.0.0 && \
-		ln -sf libip6tc.so libip6tc.so.2 && \
-		ln -sf libip6tc.so libip6tc.so.2.0.0
-	install -D iptables-1.8.x/libxtables/.libs/lib*.so $(INSTALLDIR)/iptables-1.8.x/usr/lib/
-	cd $(INSTALLDIR)/iptables-1.8.x/usr/lib && \
-		ln -sf libxtables.so libxtables.so.12 && \
-		ln -sf libxtables.so libxtables.so.12.2.0
+	install -D iptables-1.8.x/libiptc/.libs/libip4tc.so.2.0.0 $(INSTALLDIR)/iptables-1.8.x/usr/lib/libip4tc.so.2
+	install -D iptables-1.8.x/libiptc/.libs/libip6tc.so.2.0.0 $(INSTALLDIR)/iptables-1.8.x/usr/lib/libip6tc.so.2
+#	cd $(INSTALLDIR)/iptables-1.8.x/usr/lib && \
+#		ln -sf libip4tc.so libip4tc.so.2 && \
+#		ln -sf libip4tc.so libip4tc.so.2.0.0 && \
+#		ln -sf libip6tc.so libip6tc.so.2 && \
+#		ln -sf libip6tc.so libip6tc.so.2.0.0
+	install -D iptables-1.8.x/libxtables/.libs/libxtables.so.12.4.0 $(INSTALLDIR)/iptables-1.8.x/usr/lib/libxtables.so.12
+#	cd $(INSTALLDIR)/iptables-1.8.x/usr/lib && \
+#		ln -sf libxtables.so libxtables.so.12 && \
+#		ln -sf libxtables.so libxtables.so.12.2.0
 	install -D iptables-1.8.x/extensions/*.so $(INSTALLDIR)/iptables-1.8.x/usr/lib/xtables
 ifeq ($(TCONFIG_IPV6),y)
 	cd $(INSTALLDIR)/iptables-1.8.x/usr/sbin && \
@@ -1148,11 +1149,11 @@
 	@rm -f libnfnetlink/src/libnfnetlink.la
 
 libnfnetlink-install:
-	install -D libnfnetlink/src/.libs/libnfnetlink.so.0.2.0 $(INSTALLDIR)/libnfnetlink/usr/lib/libnfnetlink.so.0.2.0
-	$(STRIP) -s $(INSTALLDIR)/libnfnetlink/usr/lib/libnfnetlink.so.0.2.0
-	cd $(INSTALLDIR)/libnfnetlink/usr/lib/ && \
-		ln -sf libnfnetlink.so.0.2.0 libnfnetlink.so.0 && \
-		ln -sf libnfnetlink.so.0.2.0 libnfnetlink.so
+	install -D libnfnetlink/src/.libs/libnfnetlink.so.0.2.0 $(INSTALLDIR)/libnfnetlink/usr/lib/libnfnetlink.so.0
+	$(STRIP) -s $(INSTALLDIR)/libnfnetlink/usr/lib/libnfnetlink.so.0
+#	cd $(INSTALLDIR)/libnfnetlink/usr/lib/ && \
+#		ln -sf libnfnetlink.so.0.2.0 libnfnetlink.so.0 && \
+#		ln -sf libnfnetlink.so.0.2.0 libnfnetlink.so
 
 libnfnetlink-clean:
 	-@$(MAKE) -C libnfnetlink distclean
@@ -1241,9 +1242,9 @@
 	$(MAKE) -C $@
 
 libusb10-install:
-	install -D libusb10/libusb/.libs/libusb-1.0.so $(INSTALLDIR)/libusb10/usr/lib/libusb-1.0.so
-	$(STRIP) $(INSTALLDIR)/libusb10/usr/lib/*.so
-	cd $(INSTALLDIR)/libusb10/usr/lib && ln -sf libusb-1.0.so libusb-1.0.so.0
+	install -D libusb10/libusb/.libs/libusb-1.0.so $(INSTALLDIR)/libusb10/usr/lib/libusb-1.0.so.0
+	$(STRIP) $(INSTALLDIR)/libusb10/usr/lib/*.so.0
+#	cd $(INSTALLDIR)/libusb10/usr/lib && ln -sf libusb-1.0.so libusb-1.0.so.0
 
 libusb10-clean:
 	-@$(MAKE) -C $@ clean
@@ -1485,11 +1486,11 @@
 
 sqlite-install:
 ifeq ($(INSTALL_SQLITE),y)
-	install -D sqlite/.libs/libsqlite3.so.0.8.6 $(INSTALLDIR)/sqlite/usr/lib/libsqlite3.so.0.8.6
-	$(STRIP) $(INSTALLDIR)/sqlite/usr/lib/libsqlite3.so.0.8.6
-	cd $(INSTALLDIR)/sqlite/usr/lib/ && \
-		ln -sf libsqlite3.so.0.8.6 libsqlite3.so.0 && \
-		ln -sf libsqlite3.so.0.8.6 libsqlite3.so
+	install -D sqlite/.libs/libsqlite3.so.0.8.6 $(INSTALLDIR)/sqlite/usr/lib/libsqlite3.so.0
+	$(STRIP) $(INSTALLDIR)/sqlite/usr/lib/libsqlite3.so.0
+#	cd $(INSTALLDIR)/sqlite/usr/lib/ && \
+#		ln -sf libsqlite3.so.0.8.6 libsqlite3.so.0 && \
+#		ln -sf libsqlite3.so.0.8.6 libsqlite3.so
 endif
 	@true
 
@@ -1850,11 +1851,11 @@
 	@sed 's/#.*//g;s/[ \t]\+/ /g;s/^[ \t]*//;s/[ \t]*$$//;/^$$/d' < $(TOP)/ebtables/ethertypes > $(TARGETDIR)/rom/etc/ethertypes
 	chmod 0644 $(TARGETDIR)/rom/etc/ethertypes
 	install -D ebtables/.libs/ebtables-legacy $(INSTALLDIR)/ebtables/usr/sbin/ebtables-legacy
-	install -D ebtables/.libs/libebtc.so.0.0.0 $(INSTALLDIR)/ebtables/usr/lib/libebtc.so.0.0.0
+	install -D ebtables/.libs/libebtc.so.0.0.0 $(INSTALLDIR)/ebtables/usr/lib/libebtc.so.0
 	$(STRIP) $(INSTALLDIR)/ebtables/usr/sbin/ebtables-legacy
-	$(STRIP) $(INSTALLDIR)/ebtables/usr/lib/libebtc.so.0.0.0
+	$(STRIP) $(INSTALLDIR)/ebtables/usr/lib/libebtc.so.0
 	cd $(INSTALLDIR)/ebtables/usr/sbin/ && ln -sf ebtables-legacy ebtables
-	cd $(INSTALLDIR)/ebtables/usr/lib/ && ln -sf libebtc.so.0.0.0 libebtc.so && ln -sf libebtc.so.0.0.0 libebtc.so.0
+#	cd $(INSTALLDIR)/ebtables/usr/lib/ && ln -sf libebtc.so.0.0.0 libebtc.so && ln -sf libebtc.so.0.0.0 libebtc.so.0
 
 ebtables-clean:
 	-@$(MAKE) -C ebtables clean
@@ -1914,9 +1915,9 @@
 		$(CONFIGURE) --with-firewall=iptables --with-glib-prefix="$(INSTALLDIR)/glib/usr" --localstatedir=/var --sysconfdir=/etc
 	echo *** integrate glib to nocat installdir
 	install -d $(INSTALLDIR)/nocat/usr/lib
-	install -D glib/.libs/libglib-1.2.so.0.0.10 $(INSTALLDIR)/nocat/usr/lib/libglib-1.2.so.0.0.10
-	cd $(INSTALLDIR)/nocat/usr/lib && ln -s libglib-1.2.so.0.0.10 libglib-1.2.so.0
-	$(STRIP) $(INSTALLDIR)/nocat/usr/lib/libglib-1.2.so.0.0.10
+	install -D glib/.libs/libglib-1.2.so.0.0.10 $(INSTALLDIR)/nocat/usr/lib/libglib-1.2.so.0
+#	cd $(INSTALLDIR)/nocat/usr/lib && ln -s libglib-1.2.so.0.0.10 libglib-1.2.so.0
+	$(STRIP) $(INSTALLDIR)/nocat/usr/lib/libglib-1.2.so.0
 	@touch $@
 
 nocat: nocat/stamp-h1
@@ -1947,13 +1948,13 @@
 	@$(MAKE) -C pcre
 
 pcre-install:
-	install -D pcre/.libs/libpcre.so.1 $(INSTALLDIR)/pcre/usr/lib/libpcre.so.1.2.12
-	$(STRIP) -s $(INSTALLDIR)/pcre/usr/lib/libpcre.so.1.2.12
-	install -D pcre/.libs/libpcreposix.so.0.0.7 $(INSTALLDIR)/pcre/usr/lib/libpcreposix.so.0.0.7
-	$(STRIP) -s $(INSTALLDIR)/pcre/usr/lib/libpcreposix.so.0.0.7
-	cd $(INSTALLDIR)/pcre/usr/lib/ && \
-		ln -sf libpcre.so.1.2.12 libpcre.so.1 && \
-		ln -sf libpcreposix.so.0.0.7 libpcreposix.so.0
+	install -D pcre/.libs/libpcre.so.1 $(INSTALLDIR)/pcre/usr/lib/libpcre.so.1
+	$(STRIP) -s $(INSTALLDIR)/pcre/usr/lib/libpcre.so.1
+	install -D pcre/.libs/libpcreposix.so.0.0.7 $(INSTALLDIR)/pcre/usr/lib/libpcreposix.so.0
+	$(STRIP) -s $(INSTALLDIR)/pcre/usr/lib/libpcreposix.so.0
+#	cd $(INSTALLDIR)/pcre/usr/lib/ && \
+#		ln -sf libpcre.so.1.2.12 libpcre.so.1 && \
+#		ln -sf libpcreposix.so.0.0.7 libpcreposix.so.0
 
 pcre-clean:
 	( if [ -f pcre/Makefile ]; then \
@@ -1975,11 +1976,11 @@
 	@$(MAKE) -C libxml2 DESTDIR=$(TOP)/libxml2/staged install
 
 libxml2-install:
-	install -D libxml2/.libs/libxml2.so.2.9.12 $(INSTALLDIR)/libxml2/usr/lib/libxml2.so.2.9.12
-	$(STRIP) $(INSTALLDIR)/libxml2/usr/lib/libxml2.so.2.9.12
-	cd $(INSTALLDIR)/libxml2/usr/lib && \
-		ln -sf libxml2.so.2.9.12 libxml2.so.2 && \
-		ln -sf libxml2.so.2.9.12 libxml2.so
+	install -D libxml2/.libs/libxml2.so.2.9.12 $(INSTALLDIR)/libxml2/usr/lib/libxml2.so.2
+	$(STRIP) $(INSTALLDIR)/libxml2/usr/lib/libxml2.so.2
+#	cd $(INSTALLDIR)/libxml2/usr/lib && \
+#		ln -sf libxml2.so.2.9.12 libxml2.so.2 && \
+#		ln -sf libxml2.so.2.9.12 libxml2.so
 
 libxml2-clean:
 	-@$(MAKE) -C libxml2 clean
@@ -2000,11 +2001,11 @@
 	@rm -f $(TOP)/libpng/staged/usr/lib/libpng12.la
 
 libpng-install:
-	install -D libpng/.libs/libpng16.so.16.37.0 $(INSTALLDIR)/libpng/usr/lib/libpng16.so.16.37.0
-	$(STRIP) $(INSTALLDIR)/libpng/usr/lib/libpng16.so.16.37.0
-	cd $(INSTALLDIR)/libpng/usr/lib && \
-		ln -sf libpng16.so.16.37.0 libpng16.so && \
-		ln -sf libpng16.so.16.37.0 libpng16.so.16
+	install -D libpng/.libs/libpng16.so.16.37.0 $(INSTALLDIR)/libpng/usr/lib/libpng16.so.16
+	$(STRIP) $(INSTALLDIR)/libpng/usr/lib/libpng16.so.16
+#	cd $(INSTALLDIR)/libpng/usr/lib && \
+#		ln -sf libpng16.so.16.37.0 libpng16.so && \
+#		ln -sf libpng16.so.16.37.0 libpng16.so.16
 
 libpng-clean:
 	-@$(MAKE) -C libpng clean
@@ -2180,11 +2181,11 @@
 	$(STRIP) $(INSTALLDIR)/libncurses/usr/lib/libform.so.6
 	install libncurses/lib/libmenu.so.6.3 $(INSTALLDIR)/libncurses/usr/lib/libmenu.so.6
 	$(STRIP) $(INSTALLDIR)/libncurses/usr/lib/libmenu.so.6
-	cd $(INSTALLDIR)/libncurses/usr/lib/ && \
-		ln -sf libncurses.so.6 libncurses.so && \
-		ln -sf libpanel.so.6 libpanel.so && \
-		ln -sf libform.so.6 libform.so && \
-		ln -sf libmenu.so.6 libmenu.so
+#	cd $(INSTALLDIR)/libncurses/usr/lib/ && \
+#		ln -sf libncurses.so.6 libncurses.so && \
+#		ln -sf libpanel.so.6 libpanel.so && \
+#		ln -sf libform.so.6 libform.so && \
+#		ln -sf libmenu.so.6 libmenu.so
 
 mysql/stamp-h1: $(OPENSSLDIR) zlib libncurses
 	@cp -f $(TOP)/patches/mysql/.host.tgz $(TOP)/mysql/ || true
@@ -2248,16 +2249,16 @@
 	$(STRIP) $(INSTALLDIR)/mysql/usr/libexec/mysqld
 	cd $(INSTALLDIR)/mysql/usr/bin && ln -sf ../libexec/mysqld mysqld
 #	install -D -m 755 mysql/staged/usr/bin/mysqld_safe $(INSTALLDIR)/mysql/usr/bin
-	install -D -m 755 mysql/staged/usr/lib/mysql/libmysqlclient.so.16.0.0 $(INSTALLDIR)/mysql/usr/lib/mysql
-	$(STRIP) $(INSTALLDIR)/mysql/usr/lib/mysql/libmysqlclient.so.16.0.0
-	-@cd $(INSTALLDIR)/mysql/usr/lib/mysql && \
-		ln -sf libmysqlclient.so.16.0.0 libmysqlclient.so.16 && \
-		ln -sf libmysqlclient.so.16.0.0 libmysqlclient.so
-	install -D -m 755 mysql/staged/usr/lib/mysql/libmysqlclient_r.so.16.0.0 $(INSTALLDIR)/mysql/usr/lib/mysql
-	$(STRIP) $(INSTALLDIR)/mysql/usr/lib/mysql/libmysqlclient_r.so.16.0.0
-	-@cd $(INSTALLDIR)/mysql/usr/lib/mysql && \
-		ln -sf libmysqlclient_r.so.16.0.0 libmysqlclient_r.so.16 && \
-		ln -sf libmysqlclient_r.so.16.0.0 libmysqlclient_r.so
+	install -D -m 755 mysql/staged/usr/lib/mysql/libmysqlclient.so.16.0.0 $(INSTALLDIR)/mysql/usr/lib/mysql/libmysqlclient.so.16
+	$(STRIP) $(INSTALLDIR)/mysql/usr/lib/mysql/libmysqlclient.so.16
+#	-@cd $(INSTALLDIR)/mysql/usr/lib/mysql && \
+#		ln -sf libmysqlclient.so.16.0.0 libmysqlclient.so.16 && \
+#		ln -sf libmysqlclient.so.16.0.0 libmysqlclient.so
+	install -D -m 755 mysql/staged/usr/lib/mysql/libmysqlclient_r.so.16.0.0 $(INSTALLDIR)/mysql/usr/lib/mysql/libmysqlclient_r.so.16
+	$(STRIP) $(INSTALLDIR)/mysql/usr/lib/mysql/libmysqlclient_r.so.16
+#	-@cd $(INSTALLDIR)/mysql/usr/lib/mysql && \
+#		ln -sf libmysqlclient_r.so.16.0.0 libmysqlclient_r.so.16 && \
+#		ln -sf libmysqlclient_r.so.16.0.0 libmysqlclient_r.so
 #	-@cd $(INSTALLDIR)/mysql/usr/lib/mysql/plugin && cp -arfpu $(TOP)/mysql/staged/usr/lib/mysql/plugin/* . && \
 #	rm -f *.la *.a && \
 #	$(STRIP) *.so.*
@@ -2284,11 +2285,11 @@
 	@rm -f lzo/stamp-h1
 
 lzo-install:
-	install -D lzo/src/.libs/liblzo2.so $(INSTALLDIR)/lzo/usr/lib/liblzo2.so.2.0.0
-	$(STRIP) $(INSTALLDIR)/lzo/usr/lib/liblzo2.so.2.0.0
-	cd $(INSTALLDIR)/lzo/usr/lib && \
-		ln -sf liblzo2.so.2.0.0 liblzo2.so.2 && \
-		ln -sf liblzo2.so.2.0.0 liblzo2.so
+	install -D lzo/src/.libs/liblzo2.so $(INSTALLDIR)/lzo/usr/lib/liblzo2.so.2
+	$(STRIP) $(INSTALLDIR)/lzo/usr/lib/liblzo2.so.2
+#	cd $(INSTALLDIR)/lzo/usr/lib && \
+#		ln -sf liblzo2.so.2.0.0 liblzo2.so.2 && \
+#		ln -sf liblzo2.so.2.0.0 liblzo2.so
 
 openvpn/Makefile:
 	$(call patch_files,openvpn)
@@ -2385,9 +2386,9 @@
 	@$(MAKE) -C libcurl DESTDIR=$(TOP)/libcurl/staged install
 
 libcurl-install:
-	install -D libcurl/lib/.libs/libcurl.so.4.7.0 $(INSTALLDIR)/libcurl/usr/lib/libcurl.so.4.7.0
-	$(STRIP) -s $(INSTALLDIR)/libcurl/usr/lib/libcurl.so.4.7.0
-	cd $(INSTALLDIR)/libcurl/usr/lib/ && ln -sf libcurl.so.4.7.0 libcurl.so && ln -sf libcurl.so.4.7.0 libcurl.so.4
+	install -D libcurl/lib/.libs/libcurl.so.4.7.0 $(INSTALLDIR)/libcurl/usr/lib/libcurl.so.4
+	$(STRIP) -s $(INSTALLDIR)/libcurl/usr/lib/libcurl.so.4
+#	cd $(INSTALLDIR)/libcurl/usr/lib/ && ln -sf libcurl.so.4.7.0 libcurl.so && ln -sf libcurl.so.4.7.0 libcurl.so.4
 	install -D libcurl/src/.libs/curl $(INSTALLDIR)/libcurl/usr/sbin/curl
 	$(STRIP) -s $(INSTALLDIR)/libcurl/usr/sbin/curl
 
@@ -2440,11 +2441,11 @@
 
 libiconv-install:
 	install -d $(INSTALLDIR)/libiconv/usr/lib
-	install libiconv/lib/.libs/libiconv.so.2.6.1 $(INSTALLDIR)/libiconv/usr/lib/libiconv.so.2.6.1
-	$(STRIP) -s $(INSTALLDIR)/libiconv/usr/lib/libiconv.so.2.6.1
-	cd $(INSTALLDIR)/libiconv/usr/lib/ && \
-		ln -sf libiconv.so.2.6.1 libiconv.so.2 && \
-		ln -sf libiconv.so.2.6.1 libiconv.so
+	install libiconv/lib/.libs/libiconv.so.2.6.1 $(INSTALLDIR)/libiconv/usr/lib/libiconv.so.2
+	$(STRIP) -s $(INSTALLDIR)/libiconv/usr/lib/libiconv.so.2
+#	cd $(INSTALLDIR)/libiconv/usr/lib/ && \
+#		ln -sf libiconv.so.2.6.1 libiconv.so.2 && \
+#		ln -sf libiconv.so.2.6.1 libiconv.so
 
 transmission/build/Makefile: $(OPENSSLDIR) libcurl libevent zlib
 	@rm -rf transmission/build && mkdir -p transmission/build
@@ -2543,17 +2544,17 @@
 	install -D e2fsprogs/misc/tune2fs $(INSTALLDIR)/e2fsprogs/usr/sbin/tune2fs
 	install -D e2fsprogs/misc/badblocks $(INSTALLDIR)/e2fsprogs/usr/sbin/badblocks
 	$(STRIP) -s $(INSTALLDIR)/e2fsprogs/usr/sbin/*
-	install -D e2fsprogs/lib/libblkid.so.1.0 $(INSTALLDIR)/e2fsprogs/usr/lib/libblkid.so.1.0
-	install -D e2fsprogs/lib/libcom_err.so.2.1 $(INSTALLDIR)/e2fsprogs/usr/lib/libcom_err.so.2.1
-	install -D e2fsprogs/lib/libe2p.so.2.3 $(INSTALLDIR)/e2fsprogs/usr/lib/libe2p.so.2.3
-	install -D e2fsprogs/lib/libext2fs.so.2.4 $(INSTALLDIR)/e2fsprogs/usr/lib/libext2fs.so.2.4
-	install -D e2fsprogs/lib/libuuid.so.1.2 $(INSTALLDIR)/e2fsprogs/usr/lib/libuuid.so.1.2
+	install -D e2fsprogs/lib/libblkid.so.1.0 $(INSTALLDIR)/e2fsprogs/usr/lib/libblkid.so.1
+	install -D e2fsprogs/lib/libcom_err.so.2.1 $(INSTALLDIR)/e2fsprogs/usr/lib/libcom_err.so.2
+	install -D e2fsprogs/lib/libe2p.so.2.3 $(INSTALLDIR)/e2fsprogs/usr/lib/libe2p.so.2
+	install -D e2fsprogs/lib/libext2fs.so.2.4 $(INSTALLDIR)/e2fsprogs/usr/lib/libext2fs.so.2
+	install -D e2fsprogs/lib/libuuid.so.1.2 $(INSTALLDIR)/e2fsprogs/usr/lib/libuuid.so.1
 	$(STRIP) -s $(INSTALLDIR)/e2fsprogs/usr/lib/*.so.*
-	ln -sf libblkid.so.1.0 $(INSTALLDIR)/e2fsprogs/usr/lib/libblkid.so.1
-	ln -sf libcom_err.so.2.1 $(INSTALLDIR)/e2fsprogs/usr/lib/libcom_err.so.2
-	ln -sf libe2p.so.2.3 $(INSTALLDIR)/e2fsprogs/usr/lib/libe2p.so.2
-	ln -sf libext2fs.so.2.4 $(INSTALLDIR)/e2fsprogs/usr/lib/libext2fs.so.2
-	ln -sf libuuid.so.1.2 $(INSTALLDIR)/e2fsprogs/usr/lib/libuuid.so.1
+#	ln -sf libblkid.so.1.0 $(INSTALLDIR)/e2fsprogs/usr/lib/libblkid.so.1
+#	ln -sf libcom_err.so.2.1 $(INSTALLDIR)/e2fsprogs/usr/lib/libcom_err.so.2
+#	ln -sf libe2p.so.2.3 $(INSTALLDIR)/e2fsprogs/usr/lib/libe2p.so.2
+#	ln -sf libext2fs.so.2.4 $(INSTALLDIR)/e2fsprogs/usr/lib/libext2fs.so.2
+#	ln -sf libuuid.so.1.2 $(INSTALLDIR)/e2fsprogs/usr/lib/libuuid.so.1
 	ln -sf e2fsck $(INSTALLDIR)/e2fsprogs/usr/sbin/fsck.ext2
 	ln -sf e2fsck $(INSTALLDIR)/e2fsprogs/usr/sbin/fsck.ext3
 ifeq ($(TCONFIG_BCMARM),y)
@@ -2951,11 +2952,11 @@
 
 libmnl-install:
 	install -d $(INSTALLDIR)/libmnl/usr/lib/
-	install libmnl/src/.libs/libmnl.so.0.2.0 $(INSTALLDIR)/libmnl/usr/lib/libmnl.so.0.2.0
-	$(STRIP) $(INSTALLDIR)/libmnl/usr/lib/libmnl.so.0.2.0
-	cd $(INSTALLDIR)/libmnl/usr/lib/ && \
-		ln -sf libmnl.so.0.2.0 libmnl.so.0 && \
-		ln -sf libmnl.so.0.2.0 libmnl.so
+	install libmnl/src/.libs/libmnl.so.0.2.0 $(INSTALLDIR)/libmnl/usr/lib/libmnl.so.0
+	$(STRIP) $(INSTALLDIR)/libmnl/usr/lib/libmnl.so.0
+#	cd $(INSTALLDIR)/libmnl/usr/lib/ && \
+#		ln -sf libmnl.so.0.2.0 libmnl.so.0 && \
+#		ln -sf libmnl.so.0.2.0 libmnl.so
 
 libmnl-clean:
 	-@$(MAKE) -C libmnl clean
@@ -2979,11 +2980,11 @@
 libnetfilter_conntrack-install:
 	install -d $(INSTALLDIR)/libnetfilter_conntrack/usr/lib/
 	install libnetfilter_conntrack/src/.libs/libnetfilter_conntrack.so.3.7.0 \
-	$(INSTALLDIR)/libnetfilter_conntrack/usr/lib/libnetfilter_conntrack.so.3.7.0
-	$(STRIP) $(INSTALLDIR)/libnetfilter_conntrack/usr/lib/libnetfilter_conntrack.so.3.7.0
-	cd $(INSTALLDIR)/libnetfilter_conntrack/usr/lib/ && \
-		ln -sf libnetfilter_conntrack.so.3.7.0 libnetfilter_conntrack.so.3 && \
-		ln -sf libnetfilter_conntrack.so.3.7.0 libnetfilter_conntrack.so
+	$(INSTALLDIR)/libnetfilter_conntrack/usr/lib/libnetfilter_conntrack.so.3
+	$(STRIP) $(INSTALLDIR)/libnetfilter_conntrack/usr/lib/libnetfilter_conntrack.so.3
+#	cd $(INSTALLDIR)/libnetfilter_conntrack/usr/lib/ && \
+#		ln -sf libnetfilter_conntrack.so.3.7.0 libnetfilter_conntrack.so.3 && \
+#		ln -sf libnetfilter_conntrack.so.3.7.0 libnetfilter_conntrack.so
 
 libnetfilter_conntrack-clean:
 	-@$(MAKE) -C libnetfilter_conntrack clean
@@ -3007,11 +3008,11 @@
 libnetfilter_log-install:
 	install -d $(INSTALLDIR)/libnetfilter_log/usr/lib/
 	install libnetfilter_log/src/.libs/libnetfilter_log.so.1.1.0 \
-	$(INSTALLDIR)/libnetfilter_log/usr/lib/libnetfilter_log.so.1.1.0
-	$(STRIP) $(INSTALLDIR)/libnetfilter_log/usr/lib/libnetfilter_log.so.1.1.0
-	cd $(INSTALLDIR)/libnetfilter_log/usr/lib/ && \
-		ln -sf libnetfilter_log.so.1.1.0 libnetfilter_log.so.1 && \
-		ln -sf libnetfilter_log.so.1.1.0 libnetfilter_log.so
+	$(INSTALLDIR)/libnetfilter_log/usr/lib/libnetfilter_log.so.1
+	$(STRIP) $(INSTALLDIR)/libnetfilter_log/usr/lib/libnetfilter_log.so.1
+#	cd $(INSTALLDIR)/libnetfilter_log/usr/lib/ && \
+#		ln -sf libnetfilter_log.so.1.1.0 libnetfilter_log.so.1 && \
+#		ln -sf libnetfilter_log.so.1.1.0 libnetfilter_log.so
 
 libnetfilter_log-clean:
 	-@$(MAKE) -C libnetfilter_log clean
@@ -3035,11 +3036,11 @@
 libnetfilter_queue-install:
 	install -d $(INSTALLDIR)/libnetfilter_queue/usr/lib/
 	install libnetfilter_queue/src/.libs/libnetfilter_queue.so.1.4.0 \
-	$(INSTALLDIR)/libnetfilter_queue/usr/lib/libnetfilter_queue.so.1.4.0
-	$(STRIP) $(INSTALLDIR)/libnetfilter_queue/usr/lib/libnetfilter_queue.so.1.4.0
-	cd $(INSTALLDIR)/libnetfilter_queue/usr/lib/ && \
-		ln -sf libnetfilter_queue.so.1.4.0 libnetfilter_queue.so.1 && \
-		ln -sf libnetfilter_queue.so.1.4.0 libnetfilter_queue.so
+	$(INSTALLDIR)/libnetfilter_queue/usr/lib/libnetfilter_queue.so.1
+	$(STRIP) $(INSTALLDIR)/libnetfilter_queue/usr/lib/libnetfilter_queue.so.1
+#	cd $(INSTALLDIR)/libnetfilter_queue/usr/lib/ && \
+#		ln -sf libnetfilter_queue.so.1.4.0 libnetfilter_queue.so.1 && \
+#		ln -sf libnetfilter_queue.so.1.4.0 libnetfilter_queue.so
 
 libnetfilter_queue-clean:
 	-@$(MAKE) -C libnetfilter_queue clean
@@ -3090,12 +3091,12 @@
 ipset-install:
 	install -D ipset/src/ipset $(INSTALLDIR)/ipset/usr/sbin/ipset
 	install -d $(INSTALLDIR)/ipset/usr/lib/
-	install ipset/lib/.libs/libipset.so.11.1.0 $(INSTALLDIR)/ipset/usr/lib/libipset.so.11.1.0
-	$(STRIP) $(INSTALLDIR)/ipset/usr/lib/libipset.so.11.1.0
+	install ipset/lib/.libs/libipset.so.11.1.0 $(INSTALLDIR)/ipset/usr/lib/libipset.so.11
+	$(STRIP) $(INSTALLDIR)/ipset/usr/lib/libipset.so.11
 	$(STRIP) $(INSTALLDIR)/ipset/usr/sbin/ipset
-	cd $(INSTALLDIR)/ipset/usr/lib/ && \
-		ln -sf libipset.so.11.1.0 libipset.so.11 && \
-		ln -sf libipset.so.11.1.0 libipset.so
+#	cd $(INSTALLDIR)/ipset/usr/lib/ && \
+#		ln -sf libipset.so.11.1.0 libipset.so.11 && \
+#		ln -sf libipset.so.11.1.0 libipset.so
 
 ipset-clean:
 	-@$(MAKE) -C ipset clean
@@ -3185,12 +3186,12 @@
 iperf-install:
 	install -D iperf/src/.libs/iperf3 $(INSTALLDIR)/iperf/usr/sbin/iperf
 	install -d $(INSTALLDIR)/iperf/usr/lib/
-	install iperf/src/.libs/libiperf.so.0.0.0 $(INSTALLDIR)/iperf/usr/lib/libiperf.so.0.0.0
-	$(STRIP) $(INSTALLDIR)/iperf/usr/lib/libiperf.so.0.0.0
+	install iperf/src/.libs/libiperf.so.0.0.0 $(INSTALLDIR)/iperf/usr/lib/libiperf.so.0
+	$(STRIP) $(INSTALLDIR)/iperf/usr/lib/libiperf.so.0
 	$(STRIP) $(INSTALLDIR)/iperf/usr/sbin/iperf
-	cd $(INSTALLDIR)/iperf/usr/lib/ && \
-		ln -sf libiperf.so.0.0.0 libiperf.so.0 && \
-		ln -sf libiperf.so.0.0.0 libiperf.so
+#	cd $(INSTALLDIR)/iperf/usr/lib/ && \
+#		ln -sf libiperf.so.0.0.0 libiperf.so.0 && \
+#		ln -sf libiperf.so.0.0.0 libiperf.so
 
 iperf-clean:
 	-@$(MAKE) -C iperf clean
@@ -3234,11 +3235,11 @@
 	@$(MAKE) -C $@ DESTDIR=$(TOP)/libdaemon/staged install
 
 libdaemon-install: libdaemon
-	install -D libdaemon/libdaemon/.libs/libdaemon.so.0.5.0 $(INSTALLDIR)/libdaemon/usr/lib/libdaemon.so.0.5.0
+	install -D libdaemon/libdaemon/.libs/libdaemon.so.0.5.0 $(INSTALLDIR)/libdaemon/usr/lib/libdaemon.so.0
 	$(STRIP) $(INSTALLDIR)/libdaemon/usr/lib/*.so.*
-	cd $(INSTALLDIR)/libdaemon/usr/lib && \
-		ln -sf libdaemon.so.0.5.0 libdaemon.so && \
-		ln -sf libdaemon.so.0.5.0 libdaemon.so.0
+#	cd $(INSTALLDIR)/libdaemon/usr/lib && \
+#		ln -sf libdaemon.so.0.5.0 libdaemon.so && \
+#		ln -sf libdaemon.so.0.5.0 libdaemon.so.0
 
 libdaemon-clean:
 	-@$(MAKE) -C libdaemon distclean
@@ -3256,9 +3257,9 @@
 	@$(MAKE) -C $@ DESTDIR=$(TOP)/expat/staged install
 
 expat-install: expat
-	install -D expat/.libs/libexpat.so.1.6.2 $(INSTALLDIR)/expat/usr/lib/libexpat.so.1.6.2
-	$(STRIP) $(INSTALLDIR)/expat/usr/lib/libexpat.so.1.6.2
-	cd $(INSTALLDIR)/expat/usr/lib && ln -sf libexpat.so.1.6.2 libexpat.so.1
+	install -D expat/.libs/libexpat.so.1.6.2 $(INSTALLDIR)/expat/usr/lib/libexpat.so.1
+	$(STRIP) $(INSTALLDIR)/expat/usr/lib/libexpat.so.1
+#	cd $(INSTALLDIR)/expat/usr/lib && ln -sf libexpat.so.1.6.2 libexpat.so.1
 
 expat-clean:
 	-@$(MAKE) -C expat clean
@@ -3292,13 +3293,13 @@
 
 avahi-install: avahi
 	install -D avahi/avahi-daemon/.libs/avahi-daemon $(INSTALLDIR)/avahi/usr/sbin/avahi-daemon
-	install -D avahi/avahi-common/.libs/libavahi-common.so.3.5.3 $(INSTALLDIR)/avahi/usr/lib/libavahi-common.so.3.5.3
-	install -D avahi/avahi-core/.libs/libavahi-core.so.7.0.2 $(INSTALLDIR)/avahi/usr/lib/libavahi-core.so.7.0.2
+	install -D avahi/avahi-common/.libs/libavahi-common.so.3.5.3 $(INSTALLDIR)/avahi/usr/lib/libavahi-common.so.3
+	install -D avahi/avahi-core/.libs/libavahi-core.so.7.0.2 $(INSTALLDIR)/avahi/usr/lib/libavahi-core.so.7
 	$(STRIP) $(INSTALLDIR)/avahi/usr/sbin/avahi-daemon
-	$(STRIP) $(INSTALLDIR)/avahi/usr/lib/libavahi-common.so.3.5.3
-	$(STRIP) $(INSTALLDIR)/avahi/usr/lib/libavahi-core.so.7.0.2
-	cd $(INSTALLDIR)/avahi/usr/lib && ln -sf libavahi-common.so.3.5.3 libavahi-common.so.3
-	cd $(INSTALLDIR)/avahi/usr/lib && ln -sf libavahi-core.so.7.0.2 libavahi-core.so.7
+	$(STRIP) $(INSTALLDIR)/avahi/usr/lib/libavahi-common.so.3
+	$(STRIP) $(INSTALLDIR)/avahi/usr/lib/libavahi-core.so.7
+#	cd $(INSTALLDIR)/avahi/usr/lib && ln -sf libavahi-common.so.3.5.3 libavahi-common.so.3
+#	cd $(INSTALLDIR)/avahi/usr/lib && ln -sf libavahi-core.so.7.0.2 libavahi-core.so.7
 
 avahi-clean:
 	[ ! -f avahi/stamp-h1 ] || $(MAKE) -C avahi distclean
@@ -3318,11 +3319,11 @@
 	@$(MAKE) -C $@ DESTDIR=$(TOP)/libffi/staged install
 
 libffi-install:
-	install -D $(TOP)/libffi/.libs/libffi.so.6.0.4 $(INSTALLDIR)/libffi/usr/lib/libffi.so.6.0.4
-	$(STRIP) $(INSTALLDIR)/libffi/usr/lib/libffi.so.6.0.4
-	cd $(INSTALLDIR)/libffi/usr/lib && \
-		ln -sf libffi.so.6.0.4 libffi.so.6 && \
-		ln -sf libffi.so.6.0.4 libffi.so
+	install -D $(TOP)/libffi/.libs/libffi.so.6.0.4 $(INSTALLDIR)/libffi/usr/lib/libffi.so.6
+	$(STRIP) $(INSTALLDIR)/libffi/usr/lib/libffi.so.6
+#	cd $(INSTALLDIR)/libffi/usr/lib && \
+#		ln -sf libffi.so.6.0.4 libffi.so.6 && \
+#		ln -sf libffi.so.6.0.4 libffi.so
 
 libffi-clean:
 	[ ! -f libffi/stamp-h1 ] || $(MAKE) -C libffi distclean
@@ -3349,12 +3350,12 @@
 	@sed 's|\(\W\)/usr/lib|\1$(TOP)/glib2/staged/usr/lib|g' -i $(TOP)/glib2/staged/usr/lib/libglib-2.0.la
 
 glib2-install: glib2
-	install -D glib2/glib/.libs/libglib-2.0.so.0.3707.0 $(INSTALLDIR)/glib2/usr/lib/libglib-2.0.so.0.3707.0
-	$(STRIP) $(INSTALLDIR)/glib2/usr/lib/libglib-2.0.so.0.3707.0
+	install -D glib2/glib/.libs/libglib-2.0.so.0.3707.0 $(INSTALLDIR)/glib2/usr/lib/libglib-2.0.so.0
+	$(STRIP) $(INSTALLDIR)/glib2/usr/lib/libglib-2.0.so.0
 	-@rm -rf $(INSTALLDIR)/glib2/usr/share
-	cd $(INSTALLDIR)/glib2/usr/lib && \
-		ln -sf libglib-2.0.so.0.3707.0 libglib-2.0.so.0 && \
-		ln -sf libglib-2.0.so.0.3707.0 libglib-2.0.so
+#	cd $(INSTALLDIR)/glib2/usr/lib && \
+#		ln -sf libglib-2.0.so.0.3707.0 libglib-2.0.so.0 && \
+#		ln -sf libglib-2.0.so.0.3707.0 libglib-2.0.so
 
 glib2-clean:
 	[ ! -f glib2/stamp-h1 ] || $(MAKE) -C glib2 distclean
