--- glib.h_orig	2022-01-27 15:12:09.000000000 +0000
+++ glib.h	2021-12-31 11:30:09.000000000 +0000
@@ -1570,10 +1570,10 @@
 
 /* Bit tests
  */
-G_INLINE_FUNC gint	g_bit_nth_lsf (guint32 mask,
+/*extern gint	g_bit_nth_lsf (guint32 mask,
 				       gint    nth_bit);
 #ifdef	G_CAN_INLINE
-G_INLINE_FUNC gint
+extern inline gint
 g_bit_nth_lsf (guint32 mask,
 	       gint    nth_bit)
 {
@@ -1586,12 +1586,12 @@
   while (nth_bit < 32);
   return -1;
 }
-#endif	/* G_CAN_INLINE */
+#endif*/	/* G_CAN_INLINE */
 
-G_INLINE_FUNC gint	g_bit_nth_msf (guint32 mask,
+static inline gint	g_bit_nth_msf (guint32 mask,
 				       gint    nth_bit);
 #ifdef G_CAN_INLINE
-G_INLINE_FUNC gint
+extern inline gint
 g_bit_nth_msf (guint32 mask,
 	       gint    nth_bit)
 {
@@ -1608,7 +1608,7 @@
 }
 #endif	/* G_CAN_INLINE */
 
-G_INLINE_FUNC guint	g_bit_storage (guint number);
+/*G_INLINE_FUNC guint	g_bit_storage (guint number);
 #ifdef G_CAN_INLINE
 G_INLINE_FUNC guint
 g_bit_storage (guint number)
@@ -1623,7 +1623,7 @@
   while (number);
   return n_bits;
 }
-#endif	/* G_CAN_INLINE */
+#endif*/	/* G_CAN_INLINE */
 
 /* String Chunks
  */
