--- Makefile_lzma-loader_orig	2022-01-27 15:12:09.000000000 +0000
+++ Makefile_lzma-loader	2022-01-01 11:54:59.000000000 +0000
@@ -35,7 +35,7 @@
 		  -ffunction-sections -pipe -Wa,--trap \
 		  -mabi=32 $(REVFLAGS)
 CFLAGS		+= -DLOADADDR=$(TEXT_START) -D_LZMA_IN_CB
-CFLAGS		+= -I$(SRCBASE)/include -I$(SRCBASE)/lzma/C/Compress/Lzma
+CFLAGS		+= -I$(SRCBASE)/include -I$(SRCBASE)/lzma/C/Compress/Lzma -I$(TOOLCHAIN)/mipsel-brcm-linux-uclibc/sysroot/usr/include/sys
 
 ASFLAGS		= $(CFLAGS) -D__ASSEMBLY__ -DBZ_TEXT_START=$(BZ_TEXT_START)
 
