--- Makefile_mips_orig	2022-01-28 07:53:07.000000000 +0000
+++ Makefile	2022-01-31 10:36:08.000000000 +0000
@@ -410,7 +409,7 @@
 obj-$(TCONFIG_NGINX) += pcre
 obj-$(TCONFIG_NGINX) += libncurses
 ifneq ($(TCONFIG_BCMARM),y)
- obj-$(TCONFIG_NGINX) += libatomic_ops
+# obj-$(TCONFIG_NGINX) += libatomic_ops
 endif
 obj-$(TCONFIG_NGINX) += libiconv
 obj-$(TCONFIG_NGINX) += libxml2
@@ -629,7 +628,7 @@
 endif # !TCONFIG_BCMARM
 	cd $(TARGETDIR)/lib/modules/*/kernel/drivers/net && mv et/* . && rm -rf et || true
 	cd $(TARGETDIR)/lib/modules/*/kernel/drivers/net && mv wl/* . && rm -rf wl || true
-	cd $(TARGETDIR)/lib/modules/*/kernel/fs && mv cifs/* . && rm -rf cifs
+	cd $(TARGETDIR)/lib/modules/*/kernel/fs && mv cifs/* . && rm -rf cifs || true
 	cd $(TARGETDIR)/lib/modules/*/kernel/fs && mv jffs2/* . && rm -rf jffs2 || true
 	cd $(TARGETDIR)/lib/modules/*/kernel/fs && mv jffs/* . && rm -rf jffs || true
 	cd $(TARGETDIR)/lib/modules/*/kernel/lib && mv zlib_inflate/* . && rm -rf zlib_inflate || true
@@ -709,26 +708,28 @@
 endif
 
 # uClibc
-	install $(LIBDIR)/ld-uClibc.so.0 $(TARGETDIR)/lib/
-	install $(LIBDIR)/libcrypt.so.0 $(TARGETDIR)/lib/
-	install $(LIBDIR)/libpthread.so.0 $(TARGETDIR)/lib/
-	install $(LIBDIR)/libgcc_s.so.1 $(TARGETDIR)/lib/
+	install $(TOOLCHAIN)/mipsel-brcm-linux-uclibc/sysroot/lib/ld-uClibc.so.0 $(TARGETDIR)/lib/
+	install $(TOOLCHAIN)/mipsel-brcm-linux-uclibc/sysroot/lib/libcrypt.so.0 $(TARGETDIR)/lib/
+	install $(TOOLCHAIN)/mipsel-brcm-linux-uclibc/sysroot/lib/libpthread.so.0 $(TARGETDIR)/lib/
+	install $(TOOLCHAIN)/mipsel-brcm-linux-uclibc/sysroot/lib/libgcc_s.so.1 $(TARGETDIR)/lib/
 	$(STRIP) $(TARGETDIR)/lib/libgcc_s.so.1
-	install $(LIBDIR)/libc.so.0 $(TARGETDIR)/lib/
-	install $(LIBDIR)/libdl.so.0 $(TARGETDIR)/lib/
-	install $(LIBDIR)/libm.so.0 $(TARGETDIR)/lib/
-	install $(LIBDIR)/libnsl.so.0 $(TARGETDIR)/lib/
-	install $(LIBDIR)/libutil.so.0 $(TARGETDIR)/lib/
+	install $(TOOLCHAIN)/mipsel-brcm-linux-uclibc/sysroot/lib/libc.so.0 $(TARGETDIR)/lib/
+	install $(TOOLCHAIN)/mipsel-brcm-linux-uclibc/sysroot/lib/libdl.so.0 $(TARGETDIR)/lib/
+	install $(TOOLCHAIN)/mipsel-brcm-linux-uclibc/sysroot/lib/libm.so.0 $(TARGETDIR)/lib/
+	install $(TOOLCHAIN)/mipsel-brcm-linux-uclibc/sysroot/lib/libnsl.so.0 $(TARGETDIR)/lib/
+ifneq ($(TCONFIG_BCMARM)$(TCONFIG_SSH),)
+	install $(TOOLCHAIN)/mipsel-brcm-linux-uclibc/sysroot/lib/libutil.so.0 $(TARGETDIR)/lib/
+endif
 ifeq ($(TCONFIG_USB),y)
 ifneq ($(TCONFIG_BCMARM),y)
-	install $(LIBDIR)/librt-0.9.30.1.so $(TARGETDIR)/lib/librt.so.0
+	install $(TOOLCHAIN)/mipsel-brcm-linux-uclibc/sysroot/lib/librt.so.0 $(TARGETDIR)/lib/librt.so.0
 else
-	install $(LIBDIR)/librt.so.0 $(TARGETDIR)/lib/librt.so.0
+	install $(TOOLCHAIN)/mipsel-brcm-linux-uclibc/sysroot/lib/librt.so.0 $(TARGETDIR)/lib/librt.so.0
 endif
 endif # TCONFIG_USB
 ifneq ($(TCONFIG_NGINX)$(TCONFIG_NANO),)
 ifneq ($(TCONFIG_BCMARM),y)
-	install $(LIBDIR)/libstdc++.so.6 $(TARGETDIR)/lib/libstdc++.so.6
+	install $(TOOLCHAIN)/mipsel-brcm-linux-uclibc/lib/libstdc++.so.6 $(TARGETDIR)/lib/libstdc++.so.6
 else
 	install $(LIBDIR)/../arm-linux/lib/libstdc++.so.6 $(TARGETDIR)/lib/libstdc++.so.6
 endif # TCONFIG_BCMARM
@@ -736,7 +737,7 @@
 	$(STRIP) $(TARGETDIR)/lib/libstdc++.so.6
 endif # TCONFIG_NGINX TCONFIG_NANO
 ifneq ($(TCONFIG_OPTIMIZE_SHARED_LIBS),y)
-	install $(LIBDIR)/libresolv.so.0 $(TARGETDIR)/lib/
+	install $(TOOLCHAIN)/mipsel-brcm-linux-uclibc/sysroot/lib/libresolv.so.0 $(TARGETDIR)/lib/
 	$(STRIP) $(TARGETDIR)/lib/*.so.0
 endif
 
@@ -1841,7 +1842,7 @@
 
 sqlite/stamp-h1:
 	cd sqlite && autoreconf -fsi && \
-		CC=$(CC) CFLAGS="-Os $(EXTRACFLAGS) -ffunction-sections -fdata-sections" \
+		CC=$(CC) CFLAGS="-Os -fPIC $(EXTRACFLAGS) -ffunction-sections -fdata-sections" \
 		LDFLAGS="-ffunction-sections -fdata-sections -Wl,--gc-sections" \
 		$(CONFIGURE) --prefix=/usr --enable-shared --enable-static \
 			--disable-readline --disable-dynamic-extensions --enable-threadsafe
@@ -2102,7 +2103,7 @@
 			-I$(TOP)/sqlite -I$(TOP)/jpeg -I$(TOP)/libexif -I$(TOP)/libid3tag -I$(TOP)/libogg/include -I$(TOP)/libvorbis/include -I$(TOP)/zlib/staged/usr/include -I$(TOP)/shared" \
 		LDFLAGS="-Wl,--gc-sections -ffunction-sections -fdata-sections -L$(TOP)/libvorbis/lib/.libs -L$(TOP)/libogg/src/.libs -L$(TOP)/sqlite/.libs -L$(TOP)/libexif/libexif/.libs \
 			-L$(TOP)/jpeg -L$(TOP)/flac/src/libFLAC/.libs -L$(TOP)/libid3tag/.libs -L$(TOP)/zlib/staged/usr/lib -L$(TOP)/ffmpeg/libavformat -L$(TOP)/ffmpeg/libavcodec -L$(TOP)/ffmpeg/libavutil" \
-		LIBS="-lvorbis -logg -lsqlite3 -lexif -ljpeg -lFLAC -lid3tag -lz -lavformat -lavcodec -lavutil -lpthread -lm" \
+		LIBS="-lvorbis -logg -lsqlite3 -lexif -ljpeg -lFLAC -lid3tag -lz -lavformat -lavcodec -lavutil -lpthread -lm -lgcc_s" \
 		ac_cv_lib_id3tag__lz___id3_file_open=yes \
 		ac_cv_lib_avformat__lavcodec__lavutil__lz_avformat_open_input=no \
 		ac_cv_lib_avformat__lavcodec__lavutil__lz___av_open_input_file=yes \
@@ -2311,7 +2312,7 @@
 pcre/stamp-h1:
 	$(call patch_files,pcre)
 	cd pcre && autoreconf -fsi && \
-		CFLAGS="-Os -Wall $(EXTRACFLAGS)" \
+		CFLAGS="-Os -fPIC -Wall $(EXTRACFLAGS)" \
 		$(CONFIGURE) --prefix=/usr --disable-dependency-tracking --enable-utf8 --enable-unicode-properties --disable-cpp
 	[ -d pcre/m4 ] || mkdir pcre/m4
 	@touch $@
@@ -2363,7 +2364,7 @@
 
 libpng/stamp-h1:
 	cd libpng && \
-		CFLAGS="-Os -Wall $(EXTRACFLAGS)" \
+		CFLAGS="-Os -fPIC -Wall $(EXTRACFLAGS)" \
 		CPPFLAGS="-I$(TOP)/zlib/staged/usr/include" \
 		LDFLAGS="-L$(TOP)/zlib/staged/usr/lib" \
 		$(CONFIGURE) --prefix=/usr --enable-static --enable-shared
@@ -2500,8 +2501,8 @@
 	cd nginx && \
 		./configure --crossbuild=Linux::$(ARCH) \
 			--with-cc="$(CC)" \
-			--with-cc-opt="-Os -Wall $(EXTRACFLAGS) -I$(TOP)/$(OPENSSLDIR)/staged/usr/include -I$(TOP)/pcre -I$(TOP)/zlib/staged/usr/include $(if $(TCONFIG_BCMARM),,-I$(TOP)/libatomic_ops/src/$(comma)libs)" \
-			--with-ld-opt="-L$(TOP)/pcre/.libs -L$(TOP)/zlib/staged/usr/lib -L$(TOP)/$(OPENSSLDIR)/staged/usr/lib $(if $(TCONFIG_BCMARM),,-L$(TOP)/libatomic_ops/src)" \
+			--with-cc-opt="-Os -Wall $(EXTRACFLAGS) -I$(TOP)/$(OPENSSLDIR)/staged/usr/include -I$(TOP)/pcre -I$(TOP)/zlib/staged/usr/include" \
+			--with-ld-opt="-L$(TOP)/pcre/.libs -L$(TOP)/zlib/staged/usr/lib -L$(TOP)/$(OPENSSLDIR)/staged/usr/lib" \
 			--prefix=/usr \
 			--sbin-path=/usr/sbin \
 			--conf-path=/etc/nginx/nginx.conf \
@@ -2519,7 +2520,6 @@
 			--with-http_gzip_static_module \
 			--with-http_v2_module \
 			--with-http_realip_module \
-			$(if $(TCONFIG_BCMARM),,--with-libatomic=$(TOP)/libatomic_ops) \
 			$(if $(TCONFIG_IPV6),--with-ipv6,)
 	@touch $@
 
@@ -2619,7 +2619,7 @@
 			-DWITH_FEDERATED_STORAGE_ENGINE=FALSE \
 			-DWITH_NDBCLUSTER_STORAGE_ENGINE=FALSE \
 			-DCURSES_INCLUDE_PATH="$(TOP)/libncurses/staged/usr/include" \
-			-DCURSES_NCURSES_LIBRARY="$(TOP)/libncurses/staged/usr/lib/libncurses.so" \
+			-DCURSES_LIBRARY="$(TOP)/libncurses/staged/usr/lib/libncurses.so" \
 			-DWITH_ZLIB=system \
 			-DZLIB_INCLUDE_DIR="$(TOP)/zlib/staged/usr/include" \
 			-DZLIB_LIBRARY="$(TOP)/zlib/staged/usr/lib/libz.so" \
@@ -2952,7 +2952,7 @@
 e2fsprogs/stamp-h1:
 	$(call patch_files,e2fsprogs)
 	cd e2fsprogs && \
-		CFLAGS="-Os -Wall $(EXTRACFLAGS) -ffunction-sections -fdata-sections" \
+		CFLAGS="-Os -Wall -fPIC $(EXTRACFLAGS) -ffunction-sections -fdata-sections" \
 		LDFLAGS="-ffunction-sections -fdata-sections -Wl,--gc-sections" \
 		$(CONFIGURE) --prefix=/usr --sysconfdir=/etc --enable-elf-shlibs --disable-tls --disable-defrag $(if $(TCONFIG_BCMARM),ac_cv_lib_pthread_sem_init=no,) \
 			--disable-jbd-debug --disable-blkid-debug --disable-testio-debug --disable-backtrace --disable-e2initrd-helper \
@@ -3410,7 +3410,7 @@
 iperf/stamp-h1:
 	$(call patch_files,iperf)
 	cd iperf && autoreconf -fsi && \
-		CFLAGS="-Os -Wall $(EXTRACFLAGS)" \
+		CFLAGS="-Os -fPIC -Wall $(EXTRACFLAGS)" \
 		ac_cv_func_clock_gettime="no" \
 		ac_cv_func_daemon="no" \
 		$(CONFIGURE) --prefix=/usr --disable-profiling --without-openssl
@@ -3437,7 +3437,7 @@
 
 libdaemon/stamp-h1:
 	cd libdaemon && autoreconf -fsi && \
-		CFLAGS="-Os -Wall $(EXTRACFLAGS) -ffunction-sections -fdata-sections" \
+		CFLAGS="-Os -fPIC -Wall $(EXTRACFLAGS) -ffunction-sections -fdata-sections" \
 		LDFLAGS="-ffunction-sections -fdata-sections -Wl,--gc-sections" \
 		$(CONFIGURE) --prefix=/usr --disable-dependency-tracking ac_cv_func_setpgrp_void=yes
 	touch $@
@@ -3484,7 +3484,7 @@
 		CFLAGS="-Os -Wall $(EXTRACFLAGS) -ffunction-sections -fdata-sections -I$(TOP)/expat/lib" \
 		LDFLAGS="-L$(TOP)/expat/.libs -ldl -lpthread -ffunction-sections -fdata-sections -Wl,--gc-sections" \
 		LIBDAEMON_CFLAGS="-I$(TOP)/libdaemon" \
-		LIBDAEMON_LIBS="-L$(TOP)/libdaemon/libdaemon/.libs -ldaemon $(EXTRALDFLAGS2)" \
+		LIBDAEMON_LIBS="-L$(TOP)/libdaemon/libdaemon/.libs -ldaemon $(EXTRALDFLAGS)" \
 		$(CONFIGURE) --prefix=/usr --sysconfdir=/etc localstatedir=/var --with-distro=none \
 			--enable-introspection=no \
 			--disable-nls --disable-glib --disable-gobject \
