--- ip6tables-standalone.c.orig	2022-04-06 12:34:19.000000000 +0000
+++ ip6tables-standalone.c	2022-04-06 12:36:22.000000000 +0000
@@ -34,15 +34,12 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <errno.h>
-#include <ip6tables.h>
+#include "libiptc/libip6tc.h"
+#include "ip6tables.h"
+
 
-#ifdef IPTABLES_MULTI
-int
-ip6tables_main(int argc, char *argv[])
-#else
 int
 main(int argc, char *argv[])
-#endif
 {
 	int ret;
 	char *table = "filter";
