--- iptables-multi.c_orig	2022-01-27 15:12:09.000000000 +0000
+++ iptables-multi.c	2022-01-04 16:04:57.000000000 +0000
@@ -6,7 +6,7 @@
 int iptables_main(int argc, char **argv);
 int iptables_save_main(int argc, char **argv);
 int iptables_restore_main(int argc, char **argv);
-int iptables_xml_main(int argc, char **argv);
+//int iptables_xml_main(int argc, char **argv);
 
 int main(int argc, char **argv) {
   char *progname;
@@ -26,8 +26,8 @@
     if (!strcmp(progname, "iptables-restore"))
       return iptables_restore_main(argc, argv);
     
-    if (!strcmp(progname, "iptables-xml"))
-      return iptables_xml_main(argc, argv);
+/*    if (!strcmp(progname, "iptables-xml"))
+      return iptables_xml_main(argc, argv); */
     
     fprintf(stderr, "iptables multi-purpose version: unknown applet name %s\n", progname);
     exit(1);
