--- common.mak.orig	2022-02-27 12:42:21.820056500 +0100
+++ common.mak	2022-02-26 20:22:05.840382600 +0100
@@ -46,18 +46,23 @@
 export EXTRA_FLAG := -lgcc_s
 export ARCH := arm
 export HOST := arm-linux
+export TOOLCHAIN := $(shell cd $(dir $(shell which $(CROSS_COMPILE)strip))/.. && pwd -P)
+export LIBDIR := $(TOOLCHAIN)/$(ARCH)-brcm-linux-uclibcgnueabi/sysroot/lib
+export USRLIBDIR := $(TOOLCHAIN)/$(ARCH)-brcm-linux-uclibcgnueabi/sysroot/usr/lib
 else
 export PLATFORM := mipsel-uclibc
-export CROSS_COMPILE := mipsel-uclibc-
+export CROSS_COMPILE := mipsel-linux-
 export CROSS_COMPILER := $(CROSS_COMPILE)
 export CONFIGURE := ./configure --host=mipsel-linux --build=$(BUILD)
 export HOSTCONFIG := linux-mipsel
 export ARCH := mips
 export HOST := mipsel-linux
+export TOOLCHAIN := $(shell cd $(dir $(shell which $(CROSS_COMPILE)strip))/.. && pwd -P)
+export LIBDIR := $(TOOLCHAIN)/$(ARCH)el-brcm-linux-uclibc/sysroot/lib
+export USRLIBDIR := $(TOOLCHAIN)/$(ARCH)el-brcm-linux-uclibc/sysroot/usr/lib
 endif
 
 export PLT := $(ARCH)
-export TOOLCHAIN := $(shell cd $(dir $(shell which $(CROSS_COMPILE)strip))/.. && pwd -P)
 
 export CC := $(CROSS_COMPILE)gcc
 export CXX := $(CROSS_COMPILE)g++
@@ -97,8 +102,6 @@
 endif
 endif # TCONFIG_BCMARM
 
-export LIBDIR := $(TOOLCHAIN)/lib
-export USRLIBDIR := $(TOOLCHAIN)/usr/lib
 
 export PLATFORMDIR := $(TOP)/$(PLATFORM)
 export INSTALLDIR := $(PLATFORMDIR)/install
