--- gmessages.c.orig	2021-12-30 19:48:51.223929000 +0100
+++ gmessages.c	2021-12-30 19:36:31.933929000 +0100
@@ -113,6 +113,26 @@
 
 
 /* --- functions --- */
+G_INLINE_FUNC gint	g_bit_nth_msf (guint32 mask,
+				       gint    nth_bit);
+#ifdef G_CAN_INLINE
+G_INLINE_FUNC gint
+g_bit_nth_msf (guint32 mask,
+	       gint    nth_bit)
+{
+  if (nth_bit < 0)
+    nth_bit = 32;
+  do
+    {
+      nth_bit--;
+      if (mask & (1 << (guint) nth_bit))
+	return nth_bit;
+    }
+  while (nth_bit > 0);
+  return -1;
+}
+#endif	/* G_CAN_INLINE */
+
 static inline GLogDomain*
 g_log_find_domain (const gchar	  *log_domain)
 {
