--- string.h.orig	2021-12-30 19:54:43.433929000 +0100
+++ string.h	2021-12-23 18:36:36.535604300 +0100
@@ -124,7 +124,6 @@
 
 extern bool sysfs_streq(const char *s1, const char *s2);
 extern int strtobool(const char *s, bool *res);
-extern inline int __must_check kstrtoul(const char *s, unsigned int base, unsigned long *res);
 
 #ifdef CONFIG_BINARY_PRINTF
 int vbin_printf(u32 *bin_buf, size_t size, const char *fmt, va_list args);
